---
title: "UCSCXenaShiny v2 Book"
author: "Shensuo Li, Shixiang Wang"
date: "`r Sys.Date()`"

site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib, packages.bib]

# url: your book url like https://bookdown.org/yihui/bookdown
# cover-image: path to the social sharing image like images/cover.jpg
description: |
  This is a minimal example of using the bookdown package to write a book.
  The HTML output format for this example is bookdown::gitbook,
  set in the _output.yml file.
link-citations: yes
github-repo: rstudio/bookdown-demo
---

# About {-}

**UCSCXenaShiny** is an R package for interactively exploring UCSC Xena datasets. It is mainly designed to provide a web app (built on the top of [`{shiny}`](https://shiny.rstudio.com/) framework and [`{UCSCXenaTools}`](https://github.com/ropensci/UCSCXenaTools/) package) for downloading, analyzing and visualizing datasets from [UCSC Xena](https://xenabrowser.net/datapages/). In the latest version, the application can support both general and personalized  tumor molecular analysis with versatile functional modules and user-friendly interface layouts. Overall, our tool aims to make best use of public omics data from UCSC Xena and assist broad user community with precision oncology research.  

```{r p0000,  fig.align='center', echo = FALSE, out.width="80%"}
knitr::include_graphics('images/p0000.png')
```

Here, we have prepared the tutorial book to introduce the detailed usage of UCSCXenaShiny based on R Markdown and [bookdown](https://github.com/rstudio/bookdown). Firstly, we will start with the basic knowledge of UCSC Xena repository and its data hubs in **Chapter \@ref(intro)**. Then, the following chapters can be generally divided into two parts for the usage of R package and Shiny App:

#### Part Ⅰ {-} 

- **Chapter \@ref(install)**：The ways to install UCSCXenaShiny package.
- **Chapter \@ref(build-in)**：Avalilable built-in datasets of package, e.g. clinical or survival data.
- **Chapter \@ref(data-query)**：The functions to query molecular data of TPC datasets.
- **Chapter \@ref(data-analysis)**：The functions to perform various analyses based on TPC datasets.

#### Part Ⅱ  {-} 

- **Chapter \@ref(visit)**：The ways to visit UCSCXenaShiny application.
- **Chapter \@ref(home)**：The intuitive widgets and quick exploration modules in homepage.
- **Chapter \@ref(repo)**：The query for all datasets of UCSC Xena repository.
- **Chapter \@ref(general-da)**：General analysis modules for any matrix datasets.
- **Chapter \@ref(custom-tpc)**：Custom analysis modules for TPC datasets.
- **Chapter \@ref(per-tpc-1)**：The key preprocesses of personalized TPC pipelines. 
- **Chapter \@ref(per-tpc-2)**：The versatile analyses of personalized TPC pipelines. 
- **Chapter \@ref(download)**：Two download modules for the custom data subset. 





___

 Welcome to cite the following articles when **UCSCXenaShiny** has facilitated your study:

1. **UCSCXenaTools** —

   Wang S and Liu X. The UCSCXenaTools R package: a toolkit for accessing genomics data from UCSC Xena platform, from cancer multi-omics to single-cell RNA-seq. Journal of Open Source Software. 2019;4 40:1627.

2. **UCSCXenaShiny** **v1** —

   Wang S, Xiong Y, Zhao L, Gu K, Li Y, Zhao F, et al. UCSCXenaShiny: an R/CRAN package for interactive analysis of UCSC Xena data. Bioinformatics. 2022;38 2:527-9. doi:10.1093/bioinformatics/btab561.

3. **UCSCXenaShiny** **v2** —

   Li S, et al. UCSCXenaShiny v2: Facilitating Integrative and Personalized Oncology Omics Analysis. 2024 (In preparation)
   
   

> If  you have any question, please report to the [Github issues page](https://github.com/openbiox/UCSCXenaShiny/issues) or email to lishensuo@163.com
