<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 1 Introduction | UCSCXenaShiny v2 Book</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 1 Introduction | UCSCXenaShiny v2 Book" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 1 Introduction | UCSCXenaShiny v2 Book" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Shensuo Li, Shixiang Wang" />


<meta name="date" content="2024-03-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="install.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#ucsc-xena-datasets"><i class="fa fa-check"></i><b>1.1</b> UCSC Xena Datasets</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#tcga-abbreviations"><i class="fa fa-check"></i><b>1.2</b> TCGA abbreviations</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#pcawg-abbreviations"><i class="fa fa-check"></i><b>1.3</b> PCAWG abbreviations</a></li>
</ul></li>
<li class="part"><span><b>I R package</b></span></li>
<li class="chapter" data-level="2" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i><b>2</b> Install</a>
<ul>
<li class="chapter" data-level="2.1" data-path="install.html"><a href="install.html#cran"><i class="fa fa-check"></i><b>2.1</b> CRAN</a></li>
<li class="chapter" data-level="2.2" data-path="install.html"><a href="install.html#github"><i class="fa fa-check"></i><b>2.2</b> Github</a></li>
<li class="chapter" data-level="2.3" data-path="install.html"><a href="install.html#conda"><i class="fa fa-check"></i><b>2.3</b> Conda</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="build-in.html"><a href="build-in.html"><i class="fa fa-check"></i><b>3</b> Built-in datasets</a>
<ul>
<li class="chapter" data-level="3.1" data-path="build-in.html"><a href="build-in.html#tcga"><i class="fa fa-check"></i><b>3.1</b> TCGA</a></li>
<li class="chapter" data-level="3.2" data-path="build-in.html"><a href="build-in.html#pcawg"><i class="fa fa-check"></i><b>3.2</b> PCAWG</a></li>
<li class="chapter" data-level="3.3" data-path="build-in.html"><a href="build-in.html#ccle"><i class="fa fa-check"></i><b>3.3</b> CCLE</a></li>
<li class="chapter" data-level="3.4" data-path="build-in.html"><a href="build-in.html#zenodo"><i class="fa fa-check"></i><b>3.4</b> Zenodo</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="build-in.html"><a href="build-in.html#tumor-infiltration-estimations"><i class="fa fa-check"></i><b>3.4.1</b> tumor infiltration estimations</a></li>
<li class="chapter" data-level="3.4.2" data-path="build-in.html"><a href="build-in.html#ssgsea-pathway-activities"><i class="fa fa-check"></i><b>3.4.2</b> ssGSEA pathway activities</a></li>
<li class="chapter" data-level="3.4.3" data-path="build-in.html"><a href="build-in.html#other-tcga-annotations"><i class="fa fa-check"></i><b>3.4.3</b> other TCGA annotations</a></li>
<li class="chapter" data-level="3.4.4" data-path="build-in.html"><a href="build-in.html#identifier-repository"><i class="fa fa-check"></i><b>3.4.4</b> identifier repository</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-query.html"><a href="data-query.html"><i class="fa fa-check"></i><b>4</b> Molecular data query</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-query.html"><a href="data-query.html#tcga-query"><i class="fa fa-check"></i><b>4.1</b> TCGA query</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="data-query.html"><a href="data-query.html#get_pancan_gene_value"><i class="fa fa-check"></i><b>4.1.1</b> get_pancan_gene_value()</a></li>
<li class="chapter" data-level="4.1.2" data-path="data-query.html"><a href="data-query.html#get_pancan_transcript_value"><i class="fa fa-check"></i><b>4.1.2</b> get_pancan_transcript_value()</a></li>
<li class="chapter" data-level="4.1.3" data-path="data-query.html"><a href="data-query.html#get_pancan_protein_value"><i class="fa fa-check"></i><b>4.1.3</b> get_pancan_protein_value()</a></li>
<li class="chapter" data-level="4.1.4" data-path="data-query.html"><a href="data-query.html#get_pancan_mutation_status"><i class="fa fa-check"></i><b>4.1.4</b> get_pancan_mutation_status()</a></li>
<li class="chapter" data-level="4.1.5" data-path="data-query.html"><a href="data-query.html#get_pancan_cn_value"><i class="fa fa-check"></i><b>4.1.5</b> get_pancan_cn_value()</a></li>
<li class="chapter" data-level="4.1.6" data-path="data-query.html"><a href="data-query.html#get_pancan_methylation_value"><i class="fa fa-check"></i><b>4.1.6</b> get_pancan_methylation_value()</a></li>
<li class="chapter" data-level="4.1.7" data-path="data-query.html"><a href="data-query.html#get_pancan_mirna_value"><i class="fa fa-check"></i><b>4.1.7</b> get_pancan_miRNA_value()</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="data-query.html"><a href="data-query.html#pcawg-query"><i class="fa fa-check"></i><b>4.2</b> PCAWG query</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="data-query.html"><a href="data-query.html#get_pcawg_gene_value"><i class="fa fa-check"></i><b>4.2.1</b> get_pcawg_gene_value()</a></li>
<li class="chapter" data-level="4.2.2" data-path="data-query.html"><a href="data-query.html#get_pcawg_fusion_value"><i class="fa fa-check"></i><b>4.2.2</b> get_pcawg_fusion_value()</a></li>
<li class="chapter" data-level="4.2.3" data-path="data-query.html"><a href="data-query.html#get_pcawg_mirna_value"><i class="fa fa-check"></i><b>4.2.3</b> get_pcawg_miRNA_value()</a></li>
<li class="chapter" data-level="4.2.4" data-path="data-query.html"><a href="data-query.html#get_pcawg_promoter_value"><i class="fa fa-check"></i><b>4.2.4</b> get_pcawg_promoter_value()</a></li>
<li class="chapter" data-level="4.2.5" data-path="data-query.html"><a href="data-query.html#get_pcawg_apobec_mutagenesis_value"><i class="fa fa-check"></i><b>4.2.5</b> get_pcawg_APOBEC_mutagenesis_value()</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="data-query.html"><a href="data-query.html#ccle-query"><i class="fa fa-check"></i><b>4.3</b> CCLE query</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="data-query.html"><a href="data-query.html#get_ccle_gene_value"><i class="fa fa-check"></i><b>4.3.1</b> get_ccle_gene_value()</a></li>
<li class="chapter" data-level="4.3.2" data-path="data-query.html"><a href="data-query.html#get_ccle_gene_value-1"><i class="fa fa-check"></i><b>4.3.2</b> get_ccle_gene_value()</a></li>
<li class="chapter" data-level="4.3.3" data-path="data-query.html"><a href="data-query.html#get_ccle_mutation_status"><i class="fa fa-check"></i><b>4.3.3</b> get_ccle_mutation_status()</a></li>
<li class="chapter" data-level="4.3.4" data-path="data-query.html"><a href="data-query.html#get_ccle_cn_value"><i class="fa fa-check"></i><b>4.3.4</b> get_ccle_cn_value()</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="data-query.html"><a href="data-query.html#general-query"><i class="fa fa-check"></i><b>4.4</b> General query</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="data-query.html"><a href="data-query.html#query_pancan_value"><i class="fa fa-check"></i><b>4.4.1</b> query_pancan_value()</a></li>
<li class="chapter" data-level="4.4.2" data-path="data-query.html"><a href="data-query.html#query_molecule_value"><i class="fa fa-check"></i><b>4.4.2</b> query_molecule_value()</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-analysis.html"><a href="data-analysis.html"><i class="fa fa-check"></i><b>5</b> Molecular data analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-analysis.html"><a href="data-analysis.html#tcga-analysis"><i class="fa fa-check"></i><b>5.1</b> TCGA analysis</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="data-analysis.html"><a href="data-analysis.html#comparison-analysis"><i class="fa fa-check"></i><b>5.1.1</b> Comparison analysis</a></li>
<li class="chapter" data-level="5.1.2" data-path="data-analysis.html"><a href="data-analysis.html#correlation-analysis"><i class="fa fa-check"></i><b>5.1.2</b> Correlation analysis</a></li>
<li class="chapter" data-level="5.1.3" data-path="data-analysis.html"><a href="data-analysis.html#survival-analysis"><i class="fa fa-check"></i><b>5.1.3</b> Survival analysis</a></li>
<li class="chapter" data-level="5.1.4" data-path="data-analysis.html"><a href="data-analysis.html#dimension-reduction"><i class="fa fa-check"></i><b>5.1.4</b> Dimension reduction</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="data-analysis.html"><a href="data-analysis.html#pcawg-analysis"><i class="fa fa-check"></i><b>5.2</b> PCAWG analysis</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="data-analysis.html"><a href="data-analysis.html#comparsion-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Comparsion analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="data-analysis.html"><a href="data-analysis.html#correlation-analysis-1"><i class="fa fa-check"></i><b>5.2.2</b> Correlation analysis</a></li>
<li class="chapter" data-level="5.2.3" data-path="data-analysis.html"><a href="data-analysis.html#survival-analysis-1"><i class="fa fa-check"></i><b>5.2.3</b> Survival analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="data-analysis.html"><a href="data-analysis.html#ccle-analysis"><i class="fa fa-check"></i><b>5.3</b> CCLE analysis</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-analysis.html"><a href="data-analysis.html#comparsion-analysis-1"><i class="fa fa-check"></i><b>5.3.1</b> Comparsion analysis</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-analysis.html"><a href="data-analysis.html#correlation-analysis-2"><i class="fa fa-check"></i><b>5.3.2</b> Correlation analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="data-analysis.html"><a href="data-analysis.html#general-analysis"><i class="fa fa-check"></i><b>5.4</b> General analysis</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="data-analysis.html"><a href="data-analysis.html#comparison-analysis-1"><i class="fa fa-check"></i><b>5.4.1</b> Comparison analysis</a></li>
<li class="chapter" data-level="5.4.2" data-path="data-analysis.html"><a href="data-analysis.html#correlation-analysis-3"><i class="fa fa-check"></i><b>5.4.2</b> Correlation analysis</a></li>
<li class="chapter" data-level="5.4.3" data-path="data-analysis.html"><a href="data-analysis.html#survival-analysis-2"><i class="fa fa-check"></i><b>5.4.3</b> Survival analysis</a></li>
<li class="chapter" data-level="5.4.4" data-path="data-analysis.html"><a href="data-analysis.html#dimension-reduction-1"><i class="fa fa-check"></i><b>5.4.4</b> Dimension reduction</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Shiny App</b></span></li>
<li class="chapter" data-level="6" data-path="visit.html"><a href="visit.html"><i class="fa fa-check"></i><b>6</b> Visit</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visit.html"><a href="visit.html#server-deployments"><i class="fa fa-check"></i><b>6.1</b> Server deployments</a></li>
<li class="chapter" data-level="6.2" data-path="visit.html"><a href="visit.html#local-deployment"><i class="fa fa-check"></i><b>6.2</b> Local deployment</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="home.html"><a href="home.html"><i class="fa fa-check"></i><b>7</b> Homepage</a>
<ul>
<li class="chapter" data-level="7.1" data-path="home.html"><a href="home.html#quick-molecule-exploration"><i class="fa fa-check"></i><b>7.1</b> Quick Molecule Exploration</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="home.html"><a href="home.html#daily-gene"><i class="fa fa-check"></i><b>7.1.1</b> Daily Gene</a></li>
<li class="chapter" data-level="7.1.2" data-path="home.html"><a href="home.html#pan-cancer-query"><i class="fa fa-check"></i><b>7.1.2</b> Pan-cancer Query</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="home.html"><a href="home.html#slick-gallery"><i class="fa fa-check"></i><b>7.2</b> Slick Gallery</a></li>
<li class="chapter" data-level="7.3" data-path="home.html"><a href="home.html#tcga-analysis-guide"><i class="fa fa-check"></i><b>7.3</b> TCGA Analysis Guide</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="repo.html"><a href="repo.html"><i class="fa fa-check"></i><b>8</b> Repository</a>
<ul>
<li class="chapter" data-level="8.1" data-path="repo.html"><a href="repo.html#show-metadata"><i class="fa fa-check"></i><b>8.1</b> Show Metadata</a></li>
<li class="chapter" data-level="8.2" data-path="repo.html"><a href="repo.html#request-data"><i class="fa fa-check"></i><b>8.2</b> Request Data</a></li>
<li class="chapter" data-level="8.3" data-path="repo.html"><a href="repo.html#analyze-data"><i class="fa fa-check"></i><b>8.3</b> Analyze Data</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="general-da.html"><a href="general-da.html"><i class="fa fa-check"></i><b>9</b> General Dataset Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="general-da.html"><a href="general-da.html#m-vis-identifier-cor"><i class="fa fa-check"></i><b>9.1</b> Scatter-Correlation</a></li>
<li class="chapter" data-level="9.2" data-path="general-da.html"><a href="general-da.html#m-vis-identifier-multi-cor"><i class="fa fa-check"></i><b>9.2</b> Matrix-Correlation</a></li>
<li class="chapter" data-level="9.3" data-path="general-da.html"><a href="general-da.html#m-vis-identifier-grp-comparison"><i class="fa fa-check"></i><b>9.3</b> Group-Comparison</a></li>
<li class="chapter" data-level="9.4" data-path="general-da.html"><a href="general-da.html#m-vis-identifier-grp-surv"><i class="fa fa-check"></i><b>9.4</b> Survival-Analysis</a></li>
<li class="chapter" data-level="9.5" data-path="general-da.html"><a href="general-da.html#m-vis-identifier-dim-dist"><i class="fa fa-check"></i><b>9.5</b> Dimension-Distribution</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="custom-tpc.html"><a href="custom-tpc.html"><i class="fa fa-check"></i><b>10</b> Custom TPC modules</a>
<ul>
<li class="chapter" data-level="10.1" data-path="custom-tpc.html"><a href="custom-tpc.html#tcga-1"><i class="fa fa-check"></i><b>10.1</b> TCGA</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="custom-tpc.html"><a href="custom-tpc.html#comparison"><i class="fa fa-check"></i><b>10.1.1</b> Comparison</a></li>
<li class="chapter" data-level="10.1.2" data-path="custom-tpc.html"><a href="custom-tpc.html#correlation"><i class="fa fa-check"></i><b>10.1.2</b> Correlation</a></li>
<li class="chapter" data-level="10.1.3" data-path="custom-tpc.html"><a href="custom-tpc.html#survival-analysis-3"><i class="fa fa-check"></i><b>10.1.3</b> Survival analysis</a></li>
<li class="chapter" data-level="10.1.4" data-path="custom-tpc.html"><a href="custom-tpc.html#dimension-reduction-2"><i class="fa fa-check"></i><b>10.1.4</b> Dimension Reduction</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="custom-tpc.html"><a href="custom-tpc.html#pcawg-1"><i class="fa fa-check"></i><b>10.2</b> PCAWG</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="custom-tpc.html"><a href="custom-tpc.html#comparison-analysis-2"><i class="fa fa-check"></i><b>10.2.1</b> Comparison analysis</a></li>
<li class="chapter" data-level="10.2.2" data-path="custom-tpc.html"><a href="custom-tpc.html#correlation-analysis-4"><i class="fa fa-check"></i><b>10.2.2</b> Correlation analysis</a></li>
<li class="chapter" data-level="10.2.3" data-path="custom-tpc.html"><a href="custom-tpc.html#survival-analysis-4"><i class="fa fa-check"></i><b>10.2.3</b> Survival analysis</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="custom-tpc.html"><a href="custom-tpc.html#ccle-1"><i class="fa fa-check"></i><b>10.3</b> CCLE</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="custom-tpc.html"><a href="custom-tpc.html#comparison-analysis-3"><i class="fa fa-check"></i><b>10.3.1</b> Comparison analysis</a></li>
<li class="chapter" data-level="10.3.2" data-path="custom-tpc.html"><a href="custom-tpc.html#correlation-analysis-5"><i class="fa fa-check"></i><b>10.3.2</b> Correlation analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="per-tpc-1.html"><a href="per-tpc-1.html"><i class="fa fa-check"></i><b>11</b> Personalized TPC pipelines(1)</a>
<ul>
<li class="chapter" data-level="11.1" data-path="per-tpc-1.html"><a href="per-tpc-1.html#extensive-tumor-data"><i class="fa fa-check"></i><b>11.1</b> Extensive tumor data</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="per-tpc-1.html"><a href="per-tpc-1.html#compiling-types-of-identifiers"><i class="fa fa-check"></i><b>11.1.1</b> Compiling types of identifiers</a></li>
<li class="chapter" data-level="11.1.2" data-path="per-tpc-1.html"><a href="per-tpc-1.html#reference-of-full-identifiers"><i class="fa fa-check"></i><b>11.1.2</b> Reference of full identifiers</a></li>
<li class="chapter" data-level="11.1.3" data-path="per-tpc-1.html"><a href="per-tpc-1.html#user-defined-identifiers"><i class="fa fa-check"></i><b>11.1.3</b> User-defined identifiers</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="per-tpc-1.html"><a href="per-tpc-1.html#sample-processing"><i class="fa fa-check"></i><b>11.2</b> Sample processing</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="per-tpc-1.html"><a href="per-tpc-1.html#sample-filtering"><i class="fa fa-check"></i><b>11.2.1</b> Sample filtering</a></li>
<li class="chapter" data-level="11.2.2" data-path="per-tpc-1.html"><a href="per-tpc-1.html#sample-grouping"><i class="fa fa-check"></i><b>11.2.2</b> Sample grouping</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="per-tpc-1.html"><a href="per-tpc-1.html#data-fetch"><i class="fa fa-check"></i><b>11.3</b> Data fetch</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="per-tpc-1.html"><a href="per-tpc-1.html#single-data"><i class="fa fa-check"></i><b>11.3.1</b> Single data</a></li>
<li class="chapter" data-level="11.3.2" data-path="per-tpc-1.html"><a href="per-tpc-1.html#multiple-data"><i class="fa fa-check"></i><b>11.3.2</b> Multiple data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="per-tpc-2.html"><a href="per-tpc-2.html"><i class="fa fa-check"></i><b>12</b> Personalized TPC pipelines(2)</a>
<ul>
<li class="chapter" data-level="12.1" data-path="per-tpc-2.html"><a href="per-tpc-2.html#general-steps"><i class="fa fa-check"></i><b>12.1</b> General steps</a></li>
<li class="chapter" data-level="12.2" data-path="per-tpc-2.html"><a href="per-tpc-2.html#correlation-analysis-6"><i class="fa fa-check"></i><b>12.2</b> Correlation analysis</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="per-tpc-2.html"><a href="per-tpc-2.html#individual-mode"><i class="fa fa-check"></i><b>12.2.1</b> Individual Mode</a></li>
<li class="chapter" data-level="12.2.2" data-path="per-tpc-2.html"><a href="per-tpc-2.html#pan-cancer-mode"><i class="fa fa-check"></i><b>12.2.2</b> Pan-cancer Mode</a></li>
<li class="chapter" data-level="12.2.3" data-path="per-tpc-2.html"><a href="per-tpc-2.html#batch-screen-mode"><i class="fa fa-check"></i><b>12.2.3</b> Batch Screen Mode</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="per-tpc-2.html"><a href="per-tpc-2.html#comparison-analysis-4"><i class="fa fa-check"></i><b>12.3</b> Comparison analysis</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="per-tpc-2.html"><a href="per-tpc-2.html#individual-mode-1"><i class="fa fa-check"></i><b>12.3.1</b> Individual Mode</a></li>
<li class="chapter" data-level="12.3.2" data-path="per-tpc-2.html"><a href="per-tpc-2.html#pan-cancer-mode-1"><i class="fa fa-check"></i><b>12.3.2</b> Pan-cancer Mode</a></li>
<li class="chapter" data-level="12.3.3" data-path="per-tpc-2.html"><a href="per-tpc-2.html#batch-screen-mode-1"><i class="fa fa-check"></i><b>12.3.3</b> Batch Screen Mode</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="per-tpc-2.html"><a href="per-tpc-2.html#survival-analysis-5"><i class="fa fa-check"></i><b>12.4</b> Survival analysis</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="per-tpc-2.html"><a href="per-tpc-2.html#individual-mode-2"><i class="fa fa-check"></i><b>12.4.1</b> Individual Mode</a></li>
<li class="chapter" data-level="12.4.2" data-path="per-tpc-2.html"><a href="per-tpc-2.html#pan-cancer-mode-2"><i class="fa fa-check"></i><b>12.4.2</b> Pan-cancer Mode</a></li>
<li class="chapter" data-level="12.4.3" data-path="per-tpc-2.html"><a href="per-tpc-2.html#batch-screen-mode-2"><i class="fa fa-check"></i><b>12.4.3</b> Batch Screen Mode</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="download.html"><a href="download.html"><i class="fa fa-check"></i><b>13</b> Download modules</a>
<ul>
<li class="chapter" data-level="13.1" data-path="download.html"><a href="download.html#download-tpc-datasets"><i class="fa fa-check"></i><b>13.1</b> Download TPC datasets</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="download.html"><a href="download.html#omics-molecular-data"><i class="fa fa-check"></i><b>13.1.1</b> Omics molecular data</a></li>
<li class="chapter" data-level="13.1.2" data-path="download.html"><a href="download.html#non-omics-annotation-data"><i class="fa fa-check"></i><b>13.1.2</b> Non-omics annotation data</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="download.html"><a href="download.html#download-ucsc-xena-datasets"><i class="fa fa-check"></i><b>13.2</b> Download UCSC Xena datasets</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/openbiox/UCSCXenaShiny" target="blank">UCSCXenaShiny Github</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">UCSCXenaShiny v2 Book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="intro" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> Introduction<a href="intro.html#intro" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="ucsc-xena-datasets" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> UCSC Xena Datasets<a href="intro.html#ucsc-xena-datasets" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>UCSC Xena platform is developed by the UC Santa Cruz Genomics Institute and serves as a comprehensive repository to provide thousands of processed omics datasets from large cancer research projects (e.g. TCGA, PCAWG and CCLE) or individual research groups and enable unprecedented research opportunities.</p>
<p>The hierarchy for storing and querying datasets is as follows:</p>
<ul>
<li>The UCSC Xena repository is generally comprised of <strong>11</strong> data hubs from various portals;</li>
<li>Each data hub could include many sample cohorts;</li>
<li>Each sample cohort could involve multiple omics profiling or phenotype data;</li>
<li>Each omics profiling could have several datasets due to normalization methods or other factors.</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p0201"></span>
<img src="images/p0201.png" alt="The hierarchy of datasets in UCSC Xena" width="80%" />
<p class="caption">
Figure 1.1: The hierarchy of datasets in UCSC Xena
</p>
</div>
<blockquote>
<p>Note: In the UCSCXenaShiny, we summarized all the data (sub)types into four main types, referring to <em>clinicalMatrix</em>, <em>genomicsMatrix</em>, <em>genomicSegment</em>, <em>mutationVector</em>, where the genomicsMatrix is the key component.</p>
</blockquote>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p0202"></span>
<img src="images/p0202.png" alt="The main types and subtypes of datasets" width="80%" />
<p class="caption">
Figure 1.2: The main types and subtypes of datasets
</p>
</div>
<p>The following figure shows the numbers of cohorts and datasets for each datahub of UCSC Xena. Although some hubs (like toilHub, pancanAtlasHub) have relatively lower numbers of datasets, they mainly focus of pan-cancer integration, which could be more valuable in some research. Here, we will briefly introduce these hubs for better use.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p0203"></span>
<img src="images/p0203.png" alt="The statitics for cohorts and datasets of each data hub" width="80%" />
<p class="caption">
Figure 1.3: The statitics for cohorts and datasets of each data hub
</p>
</div>
<div id="tcgahub-tcga-hub" class="section level4 unnumbered hasAnchor">
<h4>(1) <a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Ftcga.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">tcgaHub (TCGA hub)</a><a href="intro.html#tcgahub-tcga-hub" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>Statistics</strong>: 38 cohorts, 715 datasets;</li>
<li><strong>Source</strong>: TCGA Data Coordinating Center (<a href="https://www.cancer.gov/ccg/research/genome-sequencing/tcga">DCC</a>), Jan 2016</li>
<li><strong>Description</strong>: The hub is specific to the <a href="https://www.cancer.gov/ccg/research/genome-sequencing/tcga">TCGA project</a> with both individual tumor or integrative pan-cancer cohorts</li>
</ul>
</div>
<div id="gdchub-gdc-hub" class="section level4 unnumbered hasAnchor">
<h4>(2) <a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Fgdc.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">gdcHub (GDC hub)</a><a href="intro.html#gdchub-gdc-hub" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>Statistics</strong>: 42 cohorts, 534 datasets;</li>
<li><strong>Source</strong>: GDC Data Portal (<a href="https://portal.gdc.cancer.gov/">GDC</a>), v18.0, 2019-08-28</li>
<li><strong>Description</strong>: The hub incorporates TCGA project with both individual tumor or integrative pan-cancer cohorts, as well as <a href="https://www.cancer.gov/ccg/research/genome-sequencing/target">TARGET project</a> (childhood cancers).</li>
</ul>
</div>
<div id="pancanatlashub-pan-cancer-atlas-hub" class="section level4 unnumbered hasAnchor">
<h4>(3) <a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Fpancanatlas.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">pancanAtlasHub (Pan-Cancer Atlas Hub)</a><a href="intro.html#pancanatlashub-pan-cancer-atlas-hub" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>Statistics</strong>: 1 cohort, 22 datasets;</li>
<li><strong>Source</strong>: <a href="http://www.cell.com/pb-assets/consortium/PanCancerAtlas/PanCani3/index.html">Pan-Cancer Atlas publications on Cell</a>.</li>
<li><strong>Description</strong>: The hub collects the curated pan-cancer TCGA data generated by the PanCan Atlas consortium working groups.</li>
</ul>
</div>
<div id="toilhub-ucsc-toil-rna-seq-recompute" class="section level4 unnumbered hasAnchor">
<h4>(4) <a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Ftoil.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">toilHub (UCSC Toil RNA-seq Recompute)</a><a href="intro.html#toilhub-ucsc-toil-rna-seq-recompute" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>Statistics</strong>: 5 cohort, 51 datasets;</li>
<li><strong>Source</strong>: <a href="https://www.nature.com/articles/nbt.3772">The toil pipleline from UCSC Genomics Institute</a>.</li>
<li><strong>Description</strong>: The hub aims to integrate the pan-cancer RNA-seq data from TCGA, TARGET and <a href="https://gtexportal.org/home/">GTEx</a> databases.</li>
</ul>
</div>
<div id="publichub-ucsc-public-hub" class="section level4 unnumbered hasAnchor">
<h4>(5) <a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Fucscpublic.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">publicHub (UCSC Public Hub)</a><a href="intro.html#publichub-ucsc-public-hub" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>Statistics</strong>: 37 cohort, 114 datasets;</li>
<li><strong>Source</strong>: Public resources from extensive collection.</li>
<li><strong>Description</strong>: The hub collects various cancer (or cell lines, e.g. CCLE, CMAP) omics data from other public studies.</li>
</ul>
</div>
<div id="icgchub-icgc-xena-hub" class="section level4 unnumbered hasAnchor">
<h4>(6) <a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Ficgc.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">icgcHub (ICGC Xena Hub)</a><a href="intro.html#icgchub-icgc-xena-hub" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>Statistics</strong>: 3 cohort, 23 datasets;</li>
<li><strong>Source</strong>: <a href="https://dcc.icgc.org/">International Cancer Genome Consortium(ICGC)</a>.</li>
<li><strong>Description</strong>: The hub considers the genomics data from ICGC but only included the US(TCGA) related projects for Gene/Protein expression data.</li>
</ul>
</div>
<div id="pcawghub-pcawg-hub" class="section level4 unnumbered hasAnchor">
<h4>(7) <a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Fpcawg.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">pcawgHub (PCAWG Hub)</a><a href="intro.html#pcawghub-pcawg-hub" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>Statistics</strong>: 2 cohort, 53 datasets;</li>
<li><strong>Source</strong>: <a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Fpcawg.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">The Pancancer Analysis of Whole Genomes (PCAWG) study</a>.</li>
<li><strong>Description</strong>: The hub focus on multi-omics data of cancer whole genomes across many tumor types from <a href="https://dcc.icgc.org/">International Cancer Genome Consortium(ICGC)</a>.</li>
</ul>
</div>
<div id="atacseqhubatac-seq-hub" class="section level4 unnumbered hasAnchor">
<h4>(8) <a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Fatacseq.xenahubs.net">atacseqHub(ATAC-seq Hub)</a><a href="intro.html#atacseqhubatac-seq-hub" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>Statistics</strong>: 2 cohort, 9 datasets;</li>
<li><strong>Source</strong>: <a href="https://www.science.org/doi/10.1126/science.aav1898">The chromatin accessibility landscape of primary human cancers</a></li>
<li><strong>Description</strong>: The hub describes chromatin accessibility of 410 TCGA tumor samples across 23 cancer types using the ATAC-seq technology.</li>
</ul>
</div>
<div id="kidfirsthubkids-first-xena-hub" class="section level4 unnumbered hasAnchor">
<h4>(9) <a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Fkidsfirst.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">kidfirstHub(Kids First Xena Hub)</a><a href="intro.html#kidfirsthubkids-first-xena-hub" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>Statistics</strong>: 3 cohort, 50 datasets;</li>
<li><strong>Source</strong>: <a href="https://portal.kidsfirstdrc.org/">the Gabriella Miller Kids First Pediatric Research Program</a></li>
<li><strong>Description</strong>: The hub incorporates Pediatric Brain Tumor Atlas and TARGET project.</li>
</ul>
</div>
<div id="treehousehub" class="section level4 unnumbered hasAnchor">
<h4>(10) <a href="https://xenabrowser.net/datapages/?host=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">treehouseHub</a><a href="intro.html#treehousehub" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>Statistics</strong>: 16 cohorts, 44 datasets</li>
<li><strong>Source</strong>: <a href="https://treehousegenomics.soe.ucsc.edu/public-data/">the Treehouse Childhood Cancer Initiative</a></li>
<li><strong>Description</strong>: The hub is also specific to children cancer.</li>
</ul>
</div>
<div id="singlecellhub" class="section level4 unnumbered hasAnchor">
<h4>(11) <a href="https://singlecellnew.xenahubs.net">singlecellHub</a><a href="intro.html#singlecellhub" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ul>
<li><strong>Statistics</strong>: 16 cohorts, 71 datasets</li>
<li><strong>Source</strong>: <a href="https://data.humancellatlas.org/">Human Cell Atlas (HCA)</a></li>
<li><strong>Description</strong>: The hub includes several scRNA-seq datasets from HCA, invovling cancer or normal tissues, human or mouse origin.</li>
</ul>
</div>
</div>
<div id="tcga-abbreviations" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> TCGA abbreviations<a href="intro.html#tcga-abbreviations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<table>
<caption><span id="tab:tb04-1">Table 1.1: </span>TCGA abbreviations</caption>
<colgroup>
<col width="3%" />
<col width="16%" />
<col width="80%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Abbreviation</th>
<th align="left">Name</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">LAML</td>
<td align="left">Acute Myeloid Leukemia</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">ACC</td>
<td align="left">Adrenocortical carcinoma</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="left">BLCA</td>
<td align="left">Bladder Urothelial Carcinoma</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="left">LGG</td>
<td align="left">Brain Lower Grade Glioma</td>
</tr>
<tr class="odd">
<td align="left">5</td>
<td align="left">BRCA</td>
<td align="left">Breast invasive carcinoma</td>
</tr>
<tr class="even">
<td align="left">6</td>
<td align="left">CESC</td>
<td align="left">Cervical squamous cell carcinoma and endocervical adenocarcinoma</td>
</tr>
<tr class="odd">
<td align="left">7</td>
<td align="left">CHOL</td>
<td align="left">Cholangiocarcinoma</td>
</tr>
<tr class="even">
<td align="left">8</td>
<td align="left">LCML</td>
<td align="left">Chronic Myelogenous Leukemia</td>
</tr>
<tr class="odd">
<td align="left">9</td>
<td align="left">COAD</td>
<td align="left">Colon adenocarcinoma</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="left">CNTL</td>
<td align="left">Controls</td>
</tr>
<tr class="odd">
<td align="left">11</td>
<td align="left">ESCA</td>
<td align="left">Esophageal carcinoma</td>
</tr>
<tr class="even">
<td align="left">12</td>
<td align="left">FPPP</td>
<td align="left">FFPE Pilot Phase II</td>
</tr>
<tr class="odd">
<td align="left">13</td>
<td align="left">GBM</td>
<td align="left">Glioblastoma multiforme</td>
</tr>
<tr class="even">
<td align="left">14</td>
<td align="left">HNSC</td>
<td align="left">Head and Neck squamous cell carcinoma</td>
</tr>
<tr class="odd">
<td align="left">15</td>
<td align="left">KICH</td>
<td align="left">Kidney Chromophobe</td>
</tr>
<tr class="even">
<td align="left">16</td>
<td align="left">KIRC</td>
<td align="left">Kidney renal clear cell carcinoma</td>
</tr>
<tr class="odd">
<td align="left">17</td>
<td align="left">KIRP</td>
<td align="left">Kidney renal papillary cell carcinoma</td>
</tr>
<tr class="even">
<td align="left">18</td>
<td align="left">LIHC</td>
<td align="left">Liver hepatocellular carcinoma</td>
</tr>
<tr class="odd">
<td align="left">19</td>
<td align="left">LUAD</td>
<td align="left">Lung adenocarcinoma</td>
</tr>
<tr class="even">
<td align="left">20</td>
<td align="left">LUSC</td>
<td align="left">Lung squamous cell carcinoma</td>
</tr>
<tr class="odd">
<td align="left">21</td>
<td align="left">DLBC</td>
<td align="left">Lymphoid Neoplasm Diffuse Large B-cell Lymphoma</td>
</tr>
<tr class="even">
<td align="left">22</td>
<td align="left">MESO</td>
<td align="left">Mesothelioma</td>
</tr>
<tr class="odd">
<td align="left">23</td>
<td align="left">MISC</td>
<td align="left">Miscellaneous</td>
</tr>
<tr class="even">
<td align="left">24</td>
<td align="left">OV</td>
<td align="left">Ovarian serous cystadenocarcinoma</td>
</tr>
<tr class="odd">
<td align="left">25</td>
<td align="left">PAAD</td>
<td align="left">Pancreatic adenocarcinoma</td>
</tr>
<tr class="even">
<td align="left">26</td>
<td align="left">PCPG</td>
<td align="left">Pheochromocytoma and Paraganglioma</td>
</tr>
<tr class="odd">
<td align="left">27</td>
<td align="left">PRAD</td>
<td align="left">Prostate adenocarcinoma</td>
</tr>
<tr class="even">
<td align="left">28</td>
<td align="left">READ</td>
<td align="left">Rectum adenocarcinoma</td>
</tr>
<tr class="odd">
<td align="left">29</td>
<td align="left">SARC</td>
<td align="left">Sarcoma</td>
</tr>
<tr class="even">
<td align="left">30</td>
<td align="left">SKCM</td>
<td align="left">Skin Cutaneous Melanoma</td>
</tr>
<tr class="odd">
<td align="left">31</td>
<td align="left">STAD</td>
<td align="left">Stomach adenocarcinoma</td>
</tr>
<tr class="even">
<td align="left">32</td>
<td align="left">TGCT</td>
<td align="left">Testicular Germ Cell Tumors</td>
</tr>
<tr class="odd">
<td align="left">33</td>
<td align="left">THYM</td>
<td align="left">Thymoma</td>
</tr>
<tr class="even">
<td align="left">34</td>
<td align="left">THCA</td>
<td align="left">Thyroid carcinoma</td>
</tr>
<tr class="odd">
<td align="left">35</td>
<td align="left">UCS</td>
<td align="left">Uterine Carcinosarcoma</td>
</tr>
<tr class="even">
<td align="left">36</td>
<td align="left">UCEC</td>
<td align="left">Uterine Corpus Endometrial Carcinoma</td>
</tr>
<tr class="odd">
<td align="left">37</td>
<td align="left">UVM</td>
<td align="left">Uveal Melanoma</td>
</tr>
</tbody>
</table>
</div>
<div id="pcawg-abbreviations" class="section level2 hasAnchor" number="1.3">
<h2><span class="header-section-number">1.3</span> PCAWG abbreviations<a href="intro.html#pcawg-abbreviations" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<table>
<caption><span id="tab:tb04-2">Table 1.2: </span>PCAWG abbreviations</caption>
<colgroup>
<col width="4%" />
<col width="17%" />
<col width="78%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">Abbreviation</th>
<th align="left">Name</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">BLCA-US</td>
<td align="left">Bladder Urothelial Cancer - TCGA, US</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">BRCA-US</td>
<td align="left">Breast Cancer - TCGA, US</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="left">CESC-US</td>
<td align="left">Cervical Squamous Cell Carcinoma - TCGA, US</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="left">CLLE-ES</td>
<td align="left">Chronic Lymphocytic Leukemia - ES</td>
</tr>
<tr class="odd">
<td align="left">5</td>
<td align="left">COAD-US</td>
<td align="left">Colon Adenocarcinoma - TCGA, US</td>
</tr>
<tr class="even">
<td align="left">6</td>
<td align="left">DLBC-US</td>
<td align="left">Lymphoid Neoplasm Diffuse Large B-cell Lymphoma - TCGA, US</td>
</tr>
<tr class="odd">
<td align="left">7</td>
<td align="left">ESAD-UK</td>
<td align="left">Esophageal Adenocarcinoma - UK</td>
</tr>
<tr class="even">
<td align="left">8</td>
<td align="left">GBM-US</td>
<td align="left">Brain Glioblastoma Multiforme - TCGA, US</td>
</tr>
<tr class="odd">
<td align="left">9</td>
<td align="left">HNSC-US</td>
<td align="left">Head and Neck Squamous Cell Carcinoma - TCGA, US</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="left">KICH-US</td>
<td align="left">Kidney Chromophobe - TCGA, US</td>
</tr>
<tr class="odd">
<td align="left">11</td>
<td align="left">KIRC-US</td>
<td align="left">Kidney Renal Clear Cell Carcinoma - TCGA, US</td>
</tr>
<tr class="even">
<td align="left">12</td>
<td align="left">KIRP-US</td>
<td align="left">Kidney Renal Papillary Cell Carcinoma - TCGA, US</td>
</tr>
<tr class="odd">
<td align="left">13</td>
<td align="left">LAML-US</td>
<td align="left">Acute Myeloid Leukemia - TCGA, US</td>
</tr>
<tr class="even">
<td align="left">14</td>
<td align="left">LGG-US</td>
<td align="left">Brain Lower Grade Glioma - TCGA, US</td>
</tr>
<tr class="odd">
<td align="left">15</td>
<td align="left">LIHC-US</td>
<td align="left">Liver Hepatocellular carcinoma - TCGA, US</td>
</tr>
<tr class="even">
<td align="left">16</td>
<td align="left">LIRI-JP</td>
<td align="left">Liver Cancer - RIKEN, JP</td>
</tr>
<tr class="odd">
<td align="left">17</td>
<td align="left">LUAD-US</td>
<td align="left">Lung Adenocarcinoma - TCGA, US</td>
</tr>
<tr class="even">
<td align="left">18</td>
<td align="left">LUSC-US</td>
<td align="left">Lung Squamous Cell Carcinoma - TCGA, US</td>
</tr>
<tr class="odd">
<td align="left">19</td>
<td align="left">MALY-DE</td>
<td align="left">Malignant Lymphoma - DE</td>
</tr>
<tr class="even">
<td align="left">20</td>
<td align="left">OV-AU</td>
<td align="left">Ovarian Cancer - AU</td>
</tr>
<tr class="odd">
<td align="left">21</td>
<td align="left">OV-US</td>
<td align="left">Ovarian Serous Cystadenocarcinoma - TCGA, US</td>
</tr>
<tr class="even">
<td align="left">22</td>
<td align="left">PACA-AU</td>
<td align="left">Pancreatic Cancer - AU</td>
</tr>
<tr class="odd">
<td align="left">23</td>
<td align="left">PRAD-US</td>
<td align="left">Prostate Adenocarcinoma - TCGA, US</td>
</tr>
<tr class="even">
<td align="left">24</td>
<td align="left">READ-US</td>
<td align="left">Rectum Adenocarcinoma - TCGA, US</td>
</tr>
<tr class="odd">
<td align="left">25</td>
<td align="left">RECA-EU</td>
<td align="left">Renal Cell Cancer - EU/FR</td>
</tr>
<tr class="even">
<td align="left">26</td>
<td align="left">SARC-US</td>
<td align="left">Sarcoma - TCGA, US</td>
</tr>
<tr class="odd">
<td align="left">27</td>
<td align="left">SKCM-US</td>
<td align="left">Skin Cutaneous melanoma - TCGA, US</td>
</tr>
<tr class="even">
<td align="left">28</td>
<td align="left">STAD-US</td>
<td align="left">Gastric Adenocarcinoma - TCGA, US</td>
</tr>
<tr class="odd">
<td align="left">29</td>
<td align="left">THCA-US</td>
<td align="left">Head and Neck Thyroid Carcinoma - TCGA, US</td>
</tr>
<tr class="even">
<td align="left">30</td>
<td align="left">UCEC-US</td>
<td align="left">Uterine Corpus Endometrial Carcinoma- TCGA, US</td>
</tr>
</tbody>
</table>

</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="install.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/02-background.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
