<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Molecular data analysis | UCSCXenaShiny v2 Book</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Molecular data analysis | UCSCXenaShiny v2 Book" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Molecular data analysis | UCSCXenaShiny v2 Book" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Shixiang Wang, Shensuo Li" />


<meta name="date" content="2024-02-23" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="molecular-data-query.html"/>
<link rel="next" href="visit.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About</a></li>
<li class="chapter" data-level="1" data-path="background.html"><a href="background.html"><i class="fa fa-check"></i><b>1</b> Background</a>
<ul>
<li class="chapter" data-level="1.1" data-path="background.html"><a href="background.html#ucsc-xena"><i class="fa fa-check"></i><b>1.1</b> UCSC Xena</a></li>
<li class="chapter" data-level="1.2" data-path="background.html"><a href="background.html#tcgapcawgccle"><i class="fa fa-check"></i><b>1.2</b> TCGA/PCAWG/CCLE</a></li>
<li class="chapter" data-level="1.3" data-path="background.html"><a href="background.html#other-integrated-data"><i class="fa fa-check"></i><b>1.3</b> Other integrated data</a></li>
</ul></li>
<li class="part"><span><b>I R package</b></span></li>
<li class="chapter" data-level="2" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i><b>2</b> Install</a></li>
<li class="chapter" data-level="3" data-path="built-in-datasets.html"><a href="built-in-datasets.html"><i class="fa fa-check"></i><b>3</b> Built-in datasets</a>
<ul>
<li class="chapter" data-level="3.1" data-path="built-in-datasets.html"><a href="built-in-datasets.html#tcga"><i class="fa fa-check"></i><b>3.1</b> TCGA</a></li>
<li class="chapter" data-level="3.2" data-path="built-in-datasets.html"><a href="built-in-datasets.html#pcawg"><i class="fa fa-check"></i><b>3.2</b> PCAWG</a></li>
<li class="chapter" data-level="3.3" data-path="built-in-datasets.html"><a href="built-in-datasets.html#ccle"><i class="fa fa-check"></i><b>3.3</b> CCLE</a></li>
<li class="chapter" data-level="3.4" data-path="built-in-datasets.html"><a href="built-in-datasets.html#zenodo"><i class="fa fa-check"></i><b>3.4</b> Zenodo</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="molecular-data-query.html"><a href="molecular-data-query.html"><i class="fa fa-check"></i><b>4</b> Molecular data query</a>
<ul>
<li class="chapter" data-level="4.1" data-path="molecular-data-query.html"><a href="molecular-data-query.html#tcga-query"><i class="fa fa-check"></i><b>4.1</b> TCGA query</a></li>
<li class="chapter" data-level="4.2" data-path="molecular-data-query.html"><a href="molecular-data-query.html#pcawg-query"><i class="fa fa-check"></i><b>4.2</b> PCAWG query</a></li>
<li class="chapter" data-level="4.3" data-path="molecular-data-query.html"><a href="molecular-data-query.html#ccle-query"><i class="fa fa-check"></i><b>4.3</b> CCLE query</a></li>
<li class="chapter" data-level="4.4" data-path="molecular-data-query.html"><a href="molecular-data-query.html#general-query"><i class="fa fa-check"></i><b>4.4</b> General query</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="molecular-data-analysis.html"><a href="molecular-data-analysis.html"><i class="fa fa-check"></i><b>5</b> Molecular data analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="molecular-data-analysis.html"><a href="molecular-data-analysis.html#tcga-analysis"><i class="fa fa-check"></i><b>5.1</b> TCGA analysis</a></li>
<li class="chapter" data-level="5.2" data-path="molecular-data-analysis.html"><a href="molecular-data-analysis.html#pcawg-analysis"><i class="fa fa-check"></i><b>5.2</b> PCAWG analysis</a></li>
<li class="chapter" data-level="5.3" data-path="molecular-data-analysis.html"><a href="molecular-data-analysis.html#ccle-analysis"><i class="fa fa-check"></i><b>5.3</b> CCLE analysis</a></li>
<li class="chapter" data-level="5.4" data-path="molecular-data-analysis.html"><a href="molecular-data-analysis.html#general-analysis"><i class="fa fa-check"></i><b>5.4</b> General analysis</a></li>
</ul></li>
<li class="part"><span><b>II Shiny App</b></span></li>
<li class="chapter" data-level="6" data-path="visit.html"><a href="visit.html"><i class="fa fa-check"></i><b>6</b> Visit</a></li>
<li class="chapter" data-level="7" data-path="homepage.html"><a href="homepage.html"><i class="fa fa-check"></i><b>7</b> Homepage</a>
<ul>
<li class="chapter" data-level="7.1" data-path="homepage.html"><a href="homepage.html#slick-gallery"><i class="fa fa-check"></i><b>7.1</b> Slick Gallery</a></li>
<li class="chapter" data-level="7.2" data-path="homepage.html"><a href="homepage.html#tcga-analysis-guide"><i class="fa fa-check"></i><b>7.2</b> TCGA Analysis Guide</a></li>
<li class="chapter" data-level="7.3" data-path="homepage.html"><a href="homepage.html#quick-molecule-exploration"><i class="fa fa-check"></i><b>7.3</b> Quick Molecule Exploration</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="page-repository.html"><a href="page-repository.html"><i class="fa fa-check"></i><b>8</b> Page: Repository</a></li>
<li class="chapter" data-level="9" data-path="page-quick-tcga-analysis.html"><a href="page-quick-tcga-analysis.html"><i class="fa fa-check"></i><b>9</b> Page: Quick TCGA Analysis</a></li>
<li class="chapter" data-level="10" data-path="page-personalized-tcga-analysis.html"><a href="page-personalized-tcga-analysis.html"><i class="fa fa-check"></i><b>10</b> Page: Personalized TCGA Analysis</a></li>
<li class="chapter" data-level="11" data-path="page-download.html"><a href="page-download.html"><i class="fa fa-check"></i><b>11</b> Page: Download</a></li>
<li class="chapter" data-level="12" data-path="page-help.html"><a href="page-help.html"><i class="fa fa-check"></i><b>12</b> Page: Help</a></li>
<li class="chapter" data-level="13" data-path="page-developers.html"><a href="page-developers.html"><i class="fa fa-check"></i><b>13</b> Page: Developers</a></li>
<li class="chapter" data-level="14" data-path="page-general-dataset-analysis.html"><a href="page-general-dataset-analysis.html"><i class="fa fa-check"></i><b>14</b> Page: General Dataset Analysis</a></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://github.com/openbiox/UCSCXenaShiny" target="blank">UCSCXenaShiny Github</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">UCSCXenaShiny v2 Book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="molecular-data-analysis" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Molecular data analysis<a href="molecular-data-analysis.html#molecular-data-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<ol style="list-style-type: decimal">
<li><p>For each function, we introduce its basic use and example output. Users can check all analysis or visualization parameters available by clicking the corresponding title link.</p></li>
<li><p>Among most functions, their first parameter is molecular identifier of one data type. Users can also design a molecular signature comprised of multiple molecules (e.g. <code>TP53 + 2 * KRAS - 1.3 * PTEN</code>).</p></li>
<li><p>During TPC related analysis, users can modified alternative datasets for one molecular type through the <code>opt_pancan</code> parameter. (see more via <code>str(.opt_pancan)</code>)</p></li>
</ol>
<div id="tcga-analysis" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> TCGA analysis<a href="molecular-data-analysis.html#tcga-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<table>
<caption><span id="tab:analyze-tcga-func">Table 5.1: </span>Specilized functions to analyze TCGA molecular data</caption>
<thead>
<tr class="header">
<th align="left">Database</th>
<th align="left">Type</th>
<th align="left">Function</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">TCGA</td>
<td align="left">Comparison</td>
<td align="left">vis_toil_TvsN()</td>
</tr>
<tr class="even">
<td align="left">TCGA</td>
<td align="left">Comparison</td>
<td align="left">vis_toil_TvsN_cancer()</td>
</tr>
<tr class="odd">
<td align="left">TCGA</td>
<td align="left">Comparison</td>
<td align="left">vis_toil_Mut()</td>
</tr>
<tr class="even">
<td align="left">TCGA</td>
<td align="left">Comparison</td>
<td align="left">vis_toil_Mut_cancer()</td>
</tr>
<tr class="odd">
<td align="left">TCGA</td>
<td align="left">Correlation</td>
<td align="left">vis_gene_cor()</td>
</tr>
<tr class="even">
<td align="left">TCGA</td>
<td align="left">Correlation</td>
<td align="left">vis_gene_cor_cancer()</td>
</tr>
<tr class="odd">
<td align="left">TCGA</td>
<td align="left">Correlation</td>
<td align="left">vis_gene_TIL_cor()</td>
</tr>
<tr class="even">
<td align="left">TCGA</td>
<td align="left">Correlation</td>
<td align="left">vis_gene_immune_cor()</td>
</tr>
<tr class="odd">
<td align="left">TCGA</td>
<td align="left">Correlation</td>
<td align="left">vis_gene_tmb_cor()</td>
</tr>
<tr class="even">
<td align="left">TCGA</td>
<td align="left">Correlation</td>
<td align="left">vis_gene_msi_cor()</td>
</tr>
<tr class="odd">
<td align="left">TCGA</td>
<td align="left">Correlation</td>
<td align="left">vis_gene_stemness_cor()</td>
</tr>
<tr class="even">
<td align="left">TCGA</td>
<td align="left">Correlation</td>
<td align="left">vis_gene_pw_cor()</td>
</tr>
<tr class="odd">
<td align="left">TCGA</td>
<td align="left">Survival</td>
<td align="left">tcga_surv_plot()</td>
</tr>
<tr class="even">
<td align="left">TCGA</td>
<td align="left">Survival</td>
<td align="left">vis_unicox_tree()</td>
</tr>
<tr class="odd">
<td align="left">TCGA</td>
<td align="left">Dimension Reduction</td>
<td align="left">vis_dim_dist()</td>
</tr>
</tbody>
</table>
<div id="comparison-analysis" class="section level3 hasAnchor" number="5.1.1">
<h3><span class="header-section-number">5.1.1</span> Comparison analysis<a href="molecular-data-analysis.html#comparison-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="vis_toil_tvsn" class="section level4 hasAnchor" number="5.1.1.1">
<h4><span class="header-section-number">5.1.1.1</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_toil_TvsN.html">vis_toil_TvsN()</a><a href="molecular-data-analysis.html#vis_toil_tvsn" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Compare molecular value between tumor and normal samples across pan-cancer.</p>
<ul>
<li>Basic use: <code>vis_toil_TvsN(Gene=, data_type=)</code></li>
</ul>
<div class="sourceCode" id="cb80"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb80-1"><a href="molecular-data-analysis.html#cb80-1" tabindex="-1"></a><span class="fu">vis_toil_TvsN</span>(<span class="at">Gene=</span><span class="st">&quot;TP53&quot;</span>, <span class="at">data_type =</span> <span class="st">&quot;mRNA&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-toil-TvsN"></span>
<img src="_main_files/figure-html/vis-toil-TvsN-1.png" alt="The difference of mRNA TP53 across pan-cancer" width="80%" />
<p class="caption">
Figure 5.1: The difference of mRNA TP53 across pan-cancer
</p>
</div>
<blockquote>
<p><strong>Tips</strong>: For parameter <code>data_type</code>, one of 4 molelcuar types c(“mRNA”, “transcript”, “methylation”, “miRNA”) are supported.</p>
</blockquote>
</div>
<div id="vis_toil_tvsn_cancer" class="section level4 hasAnchor" number="5.1.1.2">
<h4><span class="header-section-number">5.1.1.2</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_toil_TvsN_cancer.html">vis_toil_TvsN_cancer()</a><a href="molecular-data-analysis.html#vis_toil_tvsn_cancer" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Compare molecular value between tumor and normal samples in one cancer.</p>
<ul>
<li>Basic use: <code>vis_toil_TvsN_cancer(Gene=, data_type=, Cancer=)</code></li>
</ul>
<div class="sourceCode" id="cb81"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb81-1"><a href="molecular-data-analysis.html#cb81-1" tabindex="-1"></a><span class="fu">vis_toil_TvsN_cancer</span>(<span class="at">Gene=</span><span class="st">&quot;TP53&quot;</span>, <span class="at">data_type =</span> <span class="st">&quot;mRNA&quot;</span>, <span class="at">Cancer =</span> <span class="st">&quot;BRCA&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-toil-TvsN-cancer"></span>
<img src="_main_files/figure-html/vis-toil-TvsN-cancer-1.png" alt="The difference of mRNA TP53 in ACC cancer" width="80%" />
<p class="caption">
Figure 5.2: The difference of mRNA TP53 in ACC cancer
</p>
</div>
<blockquote>
<p><strong>Tips</strong>: For parameter <code>data_type</code>, all molelcuar types supported in function <code>query_pancan_value()</code> are applicable.</p>
</blockquote>
</div>
<div id="vis_toil_mut" class="section level4 hasAnchor" number="5.1.1.3">
<h4><span class="header-section-number">5.1.1.3</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_toil_Mut.html">vis_toil_Mut()</a><a href="molecular-data-analysis.html#vis_toil_mut" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Compare molecular value between mutation and wild tumor samples across pan-cancer.</p>
<ul>
<li>Basic use: <code>vis_toil_Mut(mut_Gene=, Gene=, data_type=)</code></li>
</ul>
<div class="sourceCode" id="cb82"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb82-1"><a href="molecular-data-analysis.html#cb82-1" tabindex="-1"></a><span class="fu">vis_toil_Mut</span>(<span class="at">mut_Gene =</span> <span class="st">&quot;TP53&quot;</span>, <span class="at">Gene =</span> <span class="st">&quot;TNF&quot;</span>, <span class="at">data_type =</span> <span class="st">&quot;mRNA&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-toil-Mut"></span>
<img src="_main_files/figure-html/vis-toil-Mut-1.png" alt="The difference of mRNA TNF between TP53-mut and TP53-wild tumor samples across pan-cancer" width="80%" />
<p class="caption">
Figure 5.3: The difference of mRNA TNF between TP53-mut and TP53-wild tumor samples across pan-cancer
</p>
</div>
<blockquote>
<p><strong>Tips</strong>: For parameter <code>data_type</code>, one of 4 molelcuar types c(“mRNA”, “transcript”, “methylation”, “miRNA”) are supported.</p>
</blockquote>
</div>
<div id="vis_toil_mut_cancer" class="section level4 hasAnchor" number="5.1.1.4">
<h4><span class="header-section-number">5.1.1.4</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_toil_Mut_cancer.html">vis_toil_Mut_cancer()</a><a href="molecular-data-analysis.html#vis_toil_mut_cancer" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Compare molecular value between mutation and wild tumor samples in one cancer.</p>
<ul>
<li>Basic use: <code>vis_toil_Mut_cancer(Gene=, data_type=, Cancer=)</code></li>
</ul>
<div class="sourceCode" id="cb83"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb83-1"><a href="molecular-data-analysis.html#cb83-1" tabindex="-1"></a><span class="fu">vis_toil_Mut_cancer</span>(<span class="at">mut_Gene =</span> <span class="st">&quot;TP53&quot;</span>, <span class="at">Gene =</span> <span class="st">&quot;TNF&quot;</span>, <span class="at">data_type =</span> <span class="st">&quot;mRNA&quot;</span>, <span class="at">Cancer =</span> <span class="st">&quot;BRCA&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-toil-Mut-cancer"></span>
<img src="_main_files/figure-html/vis-toil-Mut-cancer-1.png" alt="The difference of mRNA TNF between TP53-mut and TP53-wild tumor samples in BRCA cancer" width="80%" />
<p class="caption">
Figure 5.4: The difference of mRNA TNF between TP53-mut and TP53-wild tumor samples in BRCA cancer
</p>
</div>
<blockquote>
<p><strong>Tips</strong>: For parameter <code>data_type</code>, all molelcuar types supported in function <code>query_pancan_value()</code> are applicable.</p>
</blockquote>
</div>
</div>
<div id="correlation-analysis" class="section level3 hasAnchor" number="5.1.2">
<h3><span class="header-section-number">5.1.2</span> Correlation analysis<a href="molecular-data-analysis.html#correlation-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="vis_gene_cor" class="section level4 hasAnchor" number="5.1.2.1">
<h4><span class="header-section-number">5.1.2.1</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_gene_cor.html">vis_gene_cor()</a><a href="molecular-data-analysis.html#vis_gene_cor" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Calculate the correlation between two molecules value in tumor samples of pan-cancers.</p>
<ul>
<li>Basic use: <code>vis_gene_cor(Gene1=, data_type1=, Gene2=, data_type2=)</code></li>
</ul>
<div class="sourceCode" id="cb84"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb84-1"><a href="molecular-data-analysis.html#cb84-1" tabindex="-1"></a><span class="fu">vis_gene_cor</span>(<span class="at">Gene1 =</span> <span class="st">&quot;CSF1R&quot;</span>, <span class="at">data_type1 =</span> <span class="st">&quot;mRNA&quot;</span>, <span class="at">Gene2 =</span> <span class="st">&quot;JAK3&quot;</span>, <span class="at">data_type2 =</span> <span class="st">&quot;mRNA&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-gene-cor"></span>
<img src="_main_files/figure-html/vis-gene-cor-1.png" alt="The correlation between mRNA CSF1R and mRNA JAK3 in tumor samples of pan-cancers" width="80%" />
<p class="caption">
Figure 5.5: The correlation between mRNA CSF1R and mRNA JAK3 in tumor samples of pan-cancers
</p>
</div>
</div>
<div id="vis_gene_cor_cancer" class="section level4 hasAnchor" number="5.1.2.2">
<h4><span class="header-section-number">5.1.2.2</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_gene_cor_cancer.html">vis_gene_cor_cancer()</a><a href="molecular-data-analysis.html#vis_gene_cor_cancer" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Calculate the correlation between two molecules value in tumor samples of one cancer.</p>
<ul>
<li>Basic use: <code>vis_gene_cor_cancer(Gene1=, data_type1=, Gene2=, data_type2=, cancer_choose=)</code></li>
</ul>
<div class="sourceCode" id="cb85"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb85-1"><a href="molecular-data-analysis.html#cb85-1" tabindex="-1"></a><span class="fu">vis_gene_cor_cancer</span>(<span class="at">Gene1 =</span> <span class="st">&quot;CSF1R&quot;</span>, <span class="at">data_type1 =</span> <span class="st">&quot;mRNA&quot;</span>, </span>
<span id="cb85-2"><a href="molecular-data-analysis.html#cb85-2" tabindex="-1"></a>                    <span class="at">Gene2 =</span> <span class="st">&quot;JAK3&quot;</span>, <span class="at">data_type2 =</span> <span class="st">&quot;mRNA&quot;</span>, </span>
<span id="cb85-3"><a href="molecular-data-analysis.html#cb85-3" tabindex="-1"></a>                    <span class="at">cancer_choose =</span> <span class="st">&quot;ACC&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-gene-cor-cancer"></span>
<img src="_main_files/figure-html/vis-gene-cor-cancer-1.png" alt="The correlation between mRNA CSF1R and mRNA JAK3 in tumor samples of ACC cancer" width="80%" />
<p class="caption">
Figure 5.6: The correlation between mRNA CSF1R and mRNA JAK3 in tumor samples of ACC cancer
</p>
</div>
</div>
<div id="vis_gene_til_cor" class="section level4 hasAnchor" number="5.1.2.3">
<h4><span class="header-section-number">5.1.2.3</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_gene_TIL_cor.html">vis_gene_TIL_cor()</a><a href="molecular-data-analysis.html#vis_gene_til_cor" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Calculate the correlation between one molecule and one type of TIL in tumor samples across pan-cancers.</p>
<ul>
<li>Basic use: <code>vis_gene_TIL_cor(Gene= ,data_type= ,sig=)</code></li>
</ul>
<div class="sourceCode" id="cb86"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb86-1"><a href="molecular-data-analysis.html#cb86-1" tabindex="-1"></a>tcga_ids <span class="ot">=</span> <span class="fu">load_data</span>(<span class="st">&quot;pancan_identifier_help&quot;</span>)</span>
<span id="cb86-2"><a href="molecular-data-analysis.html#cb86-2" tabindex="-1"></a><span class="fu">names</span>(tcga_ids<span class="sc">$</span>id_TIL)</span></code></pre></div>
<pre><code>## [1] &quot;CIBERSORT&quot;     &quot;CIBERSORT-ABS&quot; &quot;EPIC&quot;          &quot;MCPCOUNTER&quot;   
## [5] &quot;QUANTISEQ&quot;     &quot;TIMER&quot;         &quot;XCELL&quot;</code></pre>
<div class="sourceCode" id="cb88"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb88-1"><a href="molecular-data-analysis.html#cb88-1" tabindex="-1"></a>sig <span class="ot">=</span> <span class="fu">paste</span>(tcga_ids<span class="sc">$</span>id_TIL<span class="sc">$</span>TIMER<span class="sc">$</span>Level3,</span>
<span id="cb88-2"><a href="molecular-data-analysis.html#cb88-2" tabindex="-1"></a>            tcga_ids<span class="sc">$</span>id_TIL<span class="sc">$</span>TIMER<span class="sc">$</span>Level2, <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb88-3"><a href="molecular-data-analysis.html#cb88-3" tabindex="-1"></a><span class="fu">vis_gene_TIL_cor</span>(<span class="at">Gene =</span> <span class="st">&quot;TP53&quot;</span>, <span class="at">data_type =</span> <span class="st">&quot;mRNA&quot;</span>,</span>
<span id="cb88-4"><a href="molecular-data-analysis.html#cb88-4" tabindex="-1"></a>                 <span class="at">sig =</span> sig)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-gene-TIL-cor"></span>
<img src="_main_files/figure-html/vis-gene-TIL-cor-1.png" alt="The correlation between mRNA TP53 and TIMER TIL in tumor samples across pan-cancers" width="1152" />
<p class="caption">
Figure 5.7: The correlation between mRNA TP53 and TIMER TIL in tumor samples across pan-cancers
</p>
</div>
</div>
<div id="vis_gene_immune_cor" class="section level4 hasAnchor" number="5.1.2.4">
<h4><span class="header-section-number">5.1.2.4</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_gene_immune_cor.html">vis_gene_immune_cor()</a><a href="molecular-data-analysis.html#vis_gene_immune_cor" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Calculate the correlation between one molecule and one type of Immune signature in tumor samples across pan-cancers.</p>
<ul>
<li>Basic use: <code>vis_gene_immune_cor(Gene= ,data_type= ,sig=)</code></li>
</ul>
<div class="sourceCode" id="cb89"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb89-1"><a href="molecular-data-analysis.html#cb89-1" tabindex="-1"></a>tcga_pan_immune_signature <span class="ot">&lt;-</span> <span class="fu">load_data</span>(<span class="st">&quot;tcga_pan_immune_signature&quot;</span>)</span></code></pre></div>
<pre><code>## Loading data from remote: https://zenodo.org/record/10554197/files/tcga_pan_immune_signature.rda, please wait...</code></pre>
<pre><code>## Data has been saved to C:/Users/xiaoxin/AppData/Local/R/win-library/4.2/UCSCXenaShiny/extdata/tcga_pan_immune_signature.rda</code></pre>
<div class="sourceCode" id="cb92"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb92-1"><a href="molecular-data-analysis.html#cb92-1" tabindex="-1"></a><span class="fu">table</span>(tcga_pan_immune_signature<span class="sc">$</span>Source)</span></code></pre></div>
<pre><code>## 
## Attractors     Bindea    c7atoms  Cibersort        ICR       Wolf      Yasin 
##          9         25         32         20          3         68          3</code></pre>
<div class="sourceCode" id="cb94"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb94-1"><a href="molecular-data-analysis.html#cb94-1" tabindex="-1"></a><span class="fu">vis_gene_immune_cor</span>(<span class="at">Gene =</span> <span class="st">&quot;TP53&quot;</span>, <span class="at">data_type =</span> <span class="st">&quot;mRNA&quot;</span>,</span>
<span id="cb94-2"><a href="molecular-data-analysis.html#cb94-2" tabindex="-1"></a>                    <span class="at">Immune_sig_type =</span> <span class="st">&quot;Cibersort&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-gene-immune-cor"></span>
<img src="_main_files/figure-html/vis-gene-immune-cor-1.png" alt="The correlation between mRNA TP53 and Cibersort signature in tumor samples across pan-cancers" width="1152" />
<p class="caption">
Figure 5.8: The correlation between mRNA TP53 and Cibersort signature in tumor samples across pan-cancers
</p>
</div>
</div>
<div id="vis_gene_tmb_cor" class="section level4 hasAnchor" number="5.1.2.5">
<h4><span class="header-section-number">5.1.2.5</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_gene_tmb_cor.html">vis_gene_tmb_cor()</a><a href="molecular-data-analysis.html#vis_gene_tmb_cor" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Calculate the correlation between one molecule and TMB score in tumor samples across pan-cancers.</p>
<ul>
<li>Basic use: <code>vis_gene_tmb_cor(Gene= , data_type= )</code></li>
</ul>
<div class="sourceCode" id="cb95"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb95-1"><a href="molecular-data-analysis.html#cb95-1" tabindex="-1"></a><span class="fu">vis_gene_tmb_cor</span>(<span class="at">Gene =</span> <span class="st">&quot;TP53&quot;</span>, <span class="at">data_type =</span> <span class="st">&quot;mRNA&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-gene-tmb-cor"></span>
<img src="_main_files/figure-html/vis-gene-tmb-cor-1.png" alt="The correlation between mRNA TP53 and TMB score in tumor samples across pan-cancers" width="1152" />
<p class="caption">
Figure 5.9: The correlation between mRNA TP53 and TMB score in tumor samples across pan-cancers
</p>
</div>
</div>
<div id="vis_gene_msi_cor" class="section level4 hasAnchor" number="5.1.2.6">
<h4><span class="header-section-number">5.1.2.6</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_gene_msi_cor.html">vis_gene_msi_cor()</a><a href="molecular-data-analysis.html#vis_gene_msi_cor" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Calculate the correlation between one molecule and MSI score in tumor samples across pan-cancers.</p>
<ul>
<li>Basic use: <code>vis_gene_msi_cor(Gene= , data_type= )</code></li>
</ul>
<div class="sourceCode" id="cb96"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb96-1"><a href="molecular-data-analysis.html#cb96-1" tabindex="-1"></a><span class="fu">vis_gene_msi_cor</span>(<span class="at">Gene =</span> <span class="st">&quot;TP53&quot;</span>, <span class="at">data_type =</span> <span class="st">&quot;mRNA&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-gene-msi-cor"></span>
<img src="_main_files/figure-html/vis-gene-msi-cor-1.png" alt="The correlation between mRNA TP53 and MSI score in tumor samples across pan-cancers" width="1152" />
<p class="caption">
Figure 5.10: The correlation between mRNA TP53 and MSI score in tumor samples across pan-cancers
</p>
</div>
</div>
<div id="vis_gene_stemness_cor" class="section level4 hasAnchor" number="5.1.2.7">
<h4><span class="header-section-number">5.1.2.7</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_gene_stemness_cor.html">vis_gene_stemness_cor()</a><a href="molecular-data-analysis.html#vis_gene_stemness_cor" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Calculate the correlation between one molecule and stemness score in tumor samples across pan-cancers.</p>
<ul>
<li>Basic use: <code>vis_gene_stemness_cor(Gene= , data_type= )</code></li>
</ul>
<div class="sourceCode" id="cb97"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb97-1"><a href="molecular-data-analysis.html#cb97-1" tabindex="-1"></a><span class="fu">vis_gene_stemness_cor</span>(<span class="at">Gene =</span> <span class="st">&quot;TP53&quot;</span>, <span class="at">data_type =</span> <span class="st">&quot;mRNA&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-gene-stemness-cor"></span>
<img src="_main_files/figure-html/vis-gene-stemness-cor-1.png" alt="The correlation between mRNA TP53 and stemness score in tumor samples across pan-cancers" width="1152" />
<p class="caption">
Figure 5.11: The correlation between mRNA TP53 and stemness score in tumor samples across pan-cancers
</p>
</div>
</div>
<div id="vis_gene_pw_cor" class="section level4 hasAnchor" number="5.1.2.8">
<h4><span class="header-section-number">5.1.2.8</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_gene_pw_cor.html">vis_gene_pw_cor()</a><a href="molecular-data-analysis.html#vis_gene_pw_cor" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Calculate the correlation between one molecule and pathway score in tumor samples of one cancer.</p>
<ul>
<li>Basic use: <code>vis_gene_pw_cor(Gene= , data_type= )</code></li>
</ul>
<div class="sourceCode" id="cb98"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb98-1"><a href="molecular-data-analysis.html#cb98-1" tabindex="-1"></a><span class="fu">vis_gene_pw_cor</span>(<span class="at">Gene =</span> <span class="st">&quot;TP53&quot;</span>, <span class="at">data_type =</span> <span class="st">&quot;mRNA&quot;</span>, </span>
<span id="cb98-2"><a href="molecular-data-analysis.html#cb98-2" tabindex="-1"></a>                <span class="at">pw_name =</span> <span class="st">&quot;HALLMARK_ADIPOGENESIS&quot;</span>,</span>
<span id="cb98-3"><a href="molecular-data-analysis.html#cb98-3" tabindex="-1"></a>                <span class="at">cancer_choose =</span> <span class="st">&quot;ACC&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-gene-pw-cor"></span>
<img src="_main_files/figure-html/vis-gene-pw-cor-1.png" alt="The correlation between mRNA TP53 and pathway score in tumor samples in ACC cancer" width="576" />
<p class="caption">
Figure 5.12: The correlation between mRNA TP53 and pathway score in tumor samples in ACC cancer
</p>
</div>
</div>
</div>
<div id="survival-analysis" class="section level3 hasAnchor" number="5.1.3">
<h3><span class="header-section-number">5.1.3</span> Survival analysis<a href="molecular-data-analysis.html#survival-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="tcga_surv_plot" class="section level4 hasAnchor" number="5.1.3.1">
<h4><span class="header-section-number">5.1.3.1</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/tcga_surv_plot.html">tcga_surv_plot()</a><a href="molecular-data-analysis.html#tcga_surv_plot" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Perform the log-rank test of one molecule for one cancer.</p>
<ul>
<li>Basic use: <code>tcga_surv_plot(data=, time= , status= )</code></li>
</ul>
<div class="sourceCode" id="cb99"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb99-1"><a href="molecular-data-analysis.html#cb99-1" tabindex="-1"></a><span class="co"># Firstly, prepare the molecular value as well as survival data</span></span>
<span id="cb99-2"><a href="molecular-data-analysis.html#cb99-2" tabindex="-1"></a>data <span class="ot">&lt;-</span> <span class="fu">tcga_surv_get</span>(<span class="at">item =</span> <span class="st">&quot;TP53&quot;</span>,<span class="at">profile =</span> <span class="st">&quot;mRNA&quot;</span>,</span>
<span id="cb99-3"><a href="molecular-data-analysis.html#cb99-3" tabindex="-1"></a>                      <span class="at">TCGA_cohort =</span> <span class="st">&quot;LUAD&quot;</span>)</span>
<span id="cb99-4"><a href="molecular-data-analysis.html#cb99-4" tabindex="-1"></a><span class="fu">head</span>(data)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 13
##   sampleID      value    OS OS.time   DSS DSS.time   DFI DFI.time   PFI PFI.time
##   &lt;chr&gt;         &lt;dbl&gt; &lt;dbl&gt;   &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;    &lt;dbl&gt;
## 1 TCGA-05-4420…  4.51     0     912     0      912     0      912     0      912
## 2 TCGA-91-6840…  5.90     0     372     0      372     0      372     0      372
## 3 TCGA-44-6778…  5.30     0    1864     0     1864     0     1864     0     1864
## 4 TCGA-67-3774…  5.22     0     385     0      385    NA       NA     0      385
## 5 TCGA-64-1679…  5.46     0    2488     0     2488     0     2488     0     2488
## 6 TCGA-55-6982…  4.54     1     995     1      995    NA       NA     1      183
## # ℹ 3 more variables: gender &lt;chr&gt;, age &lt;dbl&gt;, stage &lt;chr&gt;</code></pre>
<div class="sourceCode" id="cb101"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb101-1"><a href="molecular-data-analysis.html#cb101-1" tabindex="-1"></a><span class="fu">tcga_surv_plot</span>(data, <span class="at">time =</span> <span class="st">&quot;DSS.time&quot;</span>, <span class="at">status =</span> <span class="st">&quot;DSS&quot;</span>) <span class="co"># OS/DSS/DFI/PFI</span></span></code></pre></div>
<pre><code>## Warning in do_once((if (is_R_CMD_check()) stop else warning)(&quot;The function
## xfun::isFALSE() will be deprecated in the future. Please &quot;, : The function
## xfun::isFALSE() will be deprecated in the future. Please consider using
## base::isFALSE(x) or identical(x, FALSE) instead.</code></pre>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:tcga-surv-plot"></span>
<img src="_main_files/figure-html/tcga-surv-plot-1.png" alt="The log-rank test (DSS) of mRNA TP53 for LUAD cancer" width="576" />
<p class="caption">
Figure 5.13: The log-rank test (DSS) of mRNA TP53 for LUAD cancer
</p>
</div>
<blockquote>
<p>By default, the median data of molecular data is used to divided into two groups for log-rank test. It can be modified in corresponding paramters.</p>
</blockquote>
</div>
<div id="vis_unicox_tree" class="section level4 hasAnchor" number="5.1.3.2">
<h4><span class="header-section-number">5.1.3.2</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_unicox_tree.html">vis_unicox_tree()</a><a href="molecular-data-analysis.html#vis_unicox_tree" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Perform the Cox regression analysis of one molecule across pan-cancers.</p>
<ul>
<li>Basic use: <code>vis_unicox_tree(Gene= , data_type= , measure=)</code></li>
</ul>
<div class="sourceCode" id="cb103"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb103-1"><a href="molecular-data-analysis.html#cb103-1" tabindex="-1"></a><span class="fu">vis_unicox_tree</span>(<span class="at">Gene =</span> <span class="st">&quot;PTEN&quot;</span>, <span class="at">data_type =</span> <span class="st">&quot;mRNA&quot;</span>, <span class="at">measure =</span> <span class="st">&quot;OS&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-unicox-tree"></span>
<img src="_main_files/figure-html/vis-unicox-tree-1.png" alt="The Cox regression analysis (OS) of mRNA PTEN across pan-cancers" width="576" />
<p class="caption">
Figure 5.14: The Cox regression analysis (OS) of mRNA PTEN across pan-cancers
</p>
</div>
<blockquote>
<p>By default, the median data of molecular data is used to divided into two groups for Cox regression analysis. It can be modified in corresponding paramters.</p>
</blockquote>
</div>
</div>
<div id="dimension-reduction" class="section level3 hasAnchor" number="5.1.4">
<h3><span class="header-section-number">5.1.4</span> Dimension reduction<a href="molecular-data-analysis.html#dimension-reduction" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="vis_dim_dist" class="section level4 hasAnchor" number="5.1.4.1">
<h4><span class="header-section-number">5.1.4.1</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_dim_dist.html">vis_dim_dist()</a><a href="molecular-data-analysis.html#vis_dim_dist" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Perform dimension reduction analysis of multiple molecules for samples in groups.</p>
<ul>
<li>Basic use: <code>vis_dim_dist(ids=, data_type= ,group_info= )</code></li>
</ul>
<div class="sourceCode" id="cb104"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb104-1"><a href="molecular-data-analysis.html#cb104-1" tabindex="-1"></a><span class="co"># Firstly, prepare the grouping information of samples </span></span>
<span id="cb104-2"><a href="molecular-data-analysis.html#cb104-2" tabindex="-1"></a>group_info <span class="ot">=</span> tcga_clinical_fine <span class="sc">%&gt;%</span> </span>
<span id="cb104-3"><a href="molecular-data-analysis.html#cb104-3" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(Cancer<span class="sc">==</span><span class="st">&quot;BRCA&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb104-4"><a href="molecular-data-analysis.html#cb104-4" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Sample, Code) <span class="sc">%&gt;%</span> </span>
<span id="cb104-5"><a href="molecular-data-analysis.html#cb104-5" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">Group=</span>Code)</span>
<span id="cb104-6"><a href="molecular-data-analysis.html#cb104-6" tabindex="-1"></a><span class="fu">head</span>(group_info)</span></code></pre></div>
<pre><code>## # A tibble: 6 × 2
##   Sample          Group
##   &lt;chr&gt;           &lt;chr&gt;
## 1 TCGA-3C-AAAU-01 TP   
## 2 TCGA-3C-AALI-01 TP   
## 3 TCGA-3C-AALJ-01 TP   
## 4 TCGA-3C-AALK-01 TP   
## 5 TCGA-4H-AAAK-01 TP   
## 6 TCGA-5L-AAT0-01 TP</code></pre>
<div class="sourceCode" id="cb106"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb106-1"><a href="molecular-data-analysis.html#cb106-1" tabindex="-1"></a>ids <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;TP53&quot;</span>, <span class="st">&quot;KRAS&quot;</span>, <span class="st">&quot;PTEN&quot;</span>, <span class="st">&quot;MDM2&quot;</span>, <span class="st">&quot;CDKN1A&quot;</span>)</span>
<span id="cb106-2"><a href="molecular-data-analysis.html#cb106-2" tabindex="-1"></a><span class="fu">vis_dim_dist</span>(<span class="at">ids =</span> ids, <span class="at">data_type =</span> <span class="st">&quot;mRNA&quot;</span>, </span>
<span id="cb106-3"><a href="molecular-data-analysis.html#cb106-3" tabindex="-1"></a>             <span class="at">group_info=</span> group_info)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-dim-dist"></span>
<img src="_main_files/figure-html/vis-dim-dist-1.png" alt="The dimension reduction analysis (PCA) of 5 mRNA molcules in BRCA cancer samples grouped by tissue codes" width="576" />
<p class="caption">
Figure 5.15: The dimension reduction analysis (PCA) of 5 mRNA molcules in BRCA cancer samples grouped by tissue codes
</p>
</div>
</div>
</div>
</div>
<div id="pcawg-analysis" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> PCAWG analysis<a href="molecular-data-analysis.html#pcawg-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<table>
<caption><span id="tab:analyze-pcawg-func">Table 5.2: </span>Specilized functions to analyze PCAWG molecular data</caption>
<thead>
<tr class="header">
<th align="left">Database</th>
<th align="left">Type</th>
<th align="left">Function</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">PCAWG</td>
<td align="left">Comparison</td>
<td align="left">vis_pcawg_dist()</td>
</tr>
<tr class="even">
<td align="left">PCAWG</td>
<td align="left">Correlation</td>
<td align="left">vis_pcawg_gene_cor()</td>
</tr>
<tr class="odd">
<td align="left">PCAWG</td>
<td align="left">Survival</td>
<td align="left">vis_pcawg_unicox_tree()</td>
</tr>
</tbody>
</table>
<div id="comparsion-analysis" class="section level3 hasAnchor" number="5.2.1">
<h3><span class="header-section-number">5.2.1</span> Comparsion analysis<a href="molecular-data-analysis.html#comparsion-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="vis_pcawg_dist" class="section level4 hasAnchor" number="5.2.1.1">
<h4><span class="header-section-number">5.2.1.1</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_pcawg_dist.html">vis_pcawg_dist()</a><a href="molecular-data-analysis.html#vis_pcawg_dist" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Compare molecular value between tumor and normal samples across pan-cancer.</p>
<ul>
<li>Basic use: <code>vis_pcawg_dist(Gene= ,data_type= )</code></li>
</ul>
<div class="sourceCode" id="cb107"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb107-1"><a href="molecular-data-analysis.html#cb107-1" tabindex="-1"></a><span class="fu">vis_pcawg_dist</span>(<span class="at">Gene =</span> <span class="st">&quot;TP53&quot;</span>, <span class="at">data_type =</span> <span class="st">&quot;mRNA&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/vis-pcawg-dist-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="correlation-analysis-1" class="section level3 hasAnchor" number="5.2.2">
<h3><span class="header-section-number">5.2.2</span> Correlation analysis<a href="molecular-data-analysis.html#correlation-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="vis_pcawg_gene_cor" class="section level4 hasAnchor" number="5.2.2.1">
<h4><span class="header-section-number">5.2.2.1</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_pcawg_gene_cor.html">vis_pcawg_gene_cor()</a><a href="molecular-data-analysis.html#vis_pcawg_gene_cor" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Calculate the correlation between two molecules value in tumor samples of one cancer.</p>
<ul>
<li>Basic use: <code>vis_pcawg_gene_cor(Gene1= ,data_type1 = ,Gene2 = ,data_type2 = ,dcc_project_code_choose=)</code></li>
</ul>
<div class="sourceCode" id="cb108"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb108-1"><a href="molecular-data-analysis.html#cb108-1" tabindex="-1"></a><span class="fu">vis_pcawg_gene_cor</span>(<span class="at">Gene1 =</span> <span class="st">&quot;CSF1R&quot;</span>, <span class="at">data_type1 =</span> <span class="st">&quot;mRNA&quot;</span>,</span>
<span id="cb108-2"><a href="molecular-data-analysis.html#cb108-2" tabindex="-1"></a>                   <span class="at">Gene2 =</span> <span class="st">&quot;JAK3&quot;</span>, <span class="at">data_type2 =</span> <span class="st">&quot;mRNA&quot;</span>,</span>
<span id="cb108-3"><a href="molecular-data-analysis.html#cb108-3" tabindex="-1"></a>                   <span class="at">dcc_project_code_choose =</span> <span class="st">&quot;BLCA-US&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/vis-pcawg-gene_cor-1.png" width="80%" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="survival-analysis-1" class="section level3 hasAnchor" number="5.2.3">
<h3><span class="header-section-number">5.2.3</span> Survival analysis<a href="molecular-data-analysis.html#survival-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="vis_pcawg_unicox_tree" class="section level4 hasAnchor" number="5.2.3.1">
<h4><span class="header-section-number">5.2.3.1</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_pcawg_unicox_tree.html">vis_pcawg_unicox_tree()</a><a href="molecular-data-analysis.html#vis_pcawg_unicox_tree" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Perform the Cox regression analysis (OS) of one molecule across pan-cancers.</p>
<ul>
<li>Basic use: <code>vis_pcawg_unicox_tree(Gene= , data_type= )</code></li>
</ul>
<div class="sourceCode" id="cb109"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb109-1"><a href="molecular-data-analysis.html#cb109-1" tabindex="-1"></a><span class="fu">vis_pcawg_unicox_tree</span>(<span class="at">Gene =</span> <span class="st">&quot;TP53&quot;</span>, <span class="at">data_type =</span> <span class="st">&quot;mRNA&quot;</span>)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-pcawg-unicox-tree"></span>
<img src="_main_files/figure-html/vis-pcawg-unicox-tree-1.png" alt="The Cox regression analysis (OS) of mRNA TP53 across pan-cancers" width="576" />
<p class="caption">
Figure 5.16: The Cox regression analysis (OS) of mRNA TP53 across pan-cancers
</p>
</div>
<blockquote>
<p>By default, the median data of molecular data is used to divided into two groups for Cox regression analysis. It can be modified in corresponding paramters.</p>
</blockquote>
</div>
</div>
</div>
<div id="ccle-analysis" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> CCLE analysis<a href="molecular-data-analysis.html#ccle-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<table>
<caption><span id="tab:analyze-ccle-func">Table 5.3: </span>Specilized functions to analyze CCLE molecular data</caption>
<thead>
<tr class="header">
<th align="left">Database</th>
<th align="left">Type</th>
<th align="left">Function</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">CCLE</td>
<td align="left">Comparison</td>
<td align="left">vis_ccle_tpm()</td>
</tr>
<tr class="even">
<td align="left">CCLE</td>
<td align="left">Comparison</td>
<td align="left">vis_gene_drug_response_diff()</td>
</tr>
<tr class="odd">
<td align="left">CCLE</td>
<td align="left">Correlation</td>
<td align="left">vis_ccle_gene_cor()</td>
</tr>
<tr class="even">
<td align="left">CCLE</td>
<td align="left">Correlation</td>
<td align="left">vis_gene_drug_response_asso()</td>
</tr>
</tbody>
</table>
<div id="comparsion-analysis-1" class="section level3 hasAnchor" number="5.3.1">
<h3><span class="header-section-number">5.3.1</span> Comparsion analysis<a href="molecular-data-analysis.html#comparsion-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="vis_ccle_tpm" class="section level4 hasAnchor" number="5.3.1.1">
<h4><span class="header-section-number">5.3.1.1</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_ccle_tpm.html">vis_ccle_tpm()</a><a href="molecular-data-analysis.html#vis_ccle_tpm" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Compare molecular value among different tissues of cancer cell lines.</p>
<ul>
<li>Basic use: <code>vis_ccle_tpm(Gene= ,data_type= )</code></li>
</ul>
<div class="sourceCode" id="cb110"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb110-1"><a href="molecular-data-analysis.html#cb110-1" tabindex="-1"></a><span class="fu">vis_ccle_tpm</span>(<span class="at">Gene =</span> <span class="st">&quot;TP53&quot;</span>, <span class="at">data_type =</span> <span class="st">&quot;mRNA&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/vis-ccle-tpm-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="correlation-analysis-2" class="section level3 hasAnchor" number="5.3.2">
<h3><span class="header-section-number">5.3.2</span> Correlation analysis<a href="molecular-data-analysis.html#correlation-analysis-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="vis_ccle_gene_cor" class="section level4 hasAnchor" number="5.3.2.1">
<h4><span class="header-section-number">5.3.2.1</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_ccle_gene_cor.html">vis_ccle_gene_cor()</a><a href="molecular-data-analysis.html#vis_ccle_gene_cor" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Calculate the correlation between two molecules value in one tissue type of cancer cell lines.</p>
<ul>
<li>Basic use: <code>vis_ccle_gene_cor(Gene1= ,data_type1= ,Gene2= ,data_type2= ,SitePrimary= )</code></li>
</ul>
<div class="sourceCode" id="cb111"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb111-1"><a href="molecular-data-analysis.html#cb111-1" tabindex="-1"></a><span class="fu">vis_ccle_gene_cor</span>(<span class="at">Gene1 =</span> <span class="st">&quot;CSF1R&quot;</span>, <span class="at">data_type1 =</span> <span class="st">&quot;mRNA&quot;</span>, </span>
<span id="cb111-2"><a href="molecular-data-analysis.html#cb111-2" tabindex="-1"></a>                  <span class="at">Gene2 =</span> <span class="st">&quot;JAK3&quot;</span>, <span class="at">data_type2 =</span> <span class="st">&quot;mRNA&quot;</span>, </span>
<span id="cb111-3"><a href="molecular-data-analysis.html#cb111-3" tabindex="-1"></a>                  <span class="at">SitePrimary =</span> <span class="st">&quot;prostate&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/vis-ccle-gene_cor-1.png" width="576" style="display: block; margin: auto;" /></p>
<blockquote>
<p><code>vis_gene_drug_response_diff()</code> and <code>vis_gene_drug_response_asso()</code> are initially designed for drug pharmacogenomics analysis. In the updated shiny application, we have provided more comprehensive pharmacogenomics analysis.</p>
</blockquote>
</div>
</div>
</div>
<div id="general-analysis" class="section level2 hasAnchor" number="5.4">
<h2><span class="header-section-number">5.4</span> General analysis<a href="molecular-data-analysis.html#general-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<table>
<caption><span id="tab:analyze-general-func">Table 5.4: </span>Specilized functions to analyze General molecular data</caption>
<thead>
<tr class="header">
<th align="left">Database</th>
<th align="left">Type</th>
<th align="left">Function</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">General</td>
<td align="left">Comparison</td>
<td align="left">vis_identifier_grp_comparison()</td>
</tr>
<tr class="even">
<td align="left">General</td>
<td align="left">Correlation</td>
<td align="left">vis_identifier_cor()</td>
</tr>
<tr class="odd">
<td align="left">General</td>
<td align="left">Correlation</td>
<td align="left">vis_identifier_multi_cor()</td>
</tr>
<tr class="even">
<td align="left">General</td>
<td align="left">Survival</td>
<td align="left">vis_identifier_grp_surv()</td>
</tr>
<tr class="odd">
<td align="left">General</td>
<td align="left">Dimension Reduction</td>
<td align="left">vis_identifier_dim_dist()</td>
</tr>
</tbody>
</table>
<div id="comparison-analysis-1" class="section level3 hasAnchor" number="5.4.1">
<h3><span class="header-section-number">5.4.1</span> Comparison analysis<a href="molecular-data-analysis.html#comparison-analysis-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="vis_identifier_grp_comparison" class="section level4 hasAnchor" number="5.4.1.1">
<h4><span class="header-section-number">5.4.1.1</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_identifier_grp_comparison.html">vis_identifier_grp_comparison()</a><a href="molecular-data-analysis.html#vis_identifier_grp_comparison" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Compare molecular value between custom groups based on one genomics matrix UCSC Xena dataset.</p>
<ul>
<li>Basic use: <code>vis_identifier_grp_comparison(dataset= , id= ,grp_df= )</code></li>
</ul>
<div class="sourceCode" id="cb112"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb112-1"><a href="molecular-data-analysis.html#cb112-1" tabindex="-1"></a><span class="co"># Firstly, prepare custom groups of samples</span></span>
<span id="cb112-2"><a href="molecular-data-analysis.html#cb112-2" tabindex="-1"></a><span class="fu">library</span>(UCSCXenaTools)</span>
<span id="cb112-3"><a href="molecular-data-analysis.html#cb112-3" tabindex="-1"></a>cli_df <span class="ot">&lt;-</span> <span class="fu">XenaGenerate</span>(</span>
<span id="cb112-4"><a href="molecular-data-analysis.html#cb112-4" tabindex="-1"></a>  <span class="at">subset =</span> XenaDatasets <span class="sc">==</span> <span class="st">&quot;TCGA.LUAD.sampleMap/LUAD_clinicalMatrix&quot;</span></span>
<span id="cb112-5"><a href="molecular-data-analysis.html#cb112-5" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb112-6"><a href="molecular-data-analysis.html#cb112-6" tabindex="-1"></a>  <span class="fu">XenaQuery</span>() <span class="sc">%&gt;%</span></span>
<span id="cb112-7"><a href="molecular-data-analysis.html#cb112-7" tabindex="-1"></a>  <span class="fu">XenaDownload</span>() <span class="sc">%&gt;%</span></span>
<span id="cb112-8"><a href="molecular-data-analysis.html#cb112-8" tabindex="-1"></a>  <span class="fu">XenaPrepare</span>()</span>
<span id="cb112-9"><a href="molecular-data-analysis.html#cb112-9" tabindex="-1"></a>grp_df <span class="ot">=</span> cli_df[, <span class="fu">c</span>(<span class="st">&quot;sampleID&quot;</span>, <span class="st">&quot;pathologic_M&quot;</span>)] <span class="sc">%&gt;%</span></span>
<span id="cb112-10"><a href="molecular-data-analysis.html#cb112-10" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(pathologic_M <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&quot;M0&quot;</span>, <span class="st">&quot;M1&quot;</span>, <span class="st">&quot;MX&quot;</span>))</span>
<span id="cb112-11"><a href="molecular-data-analysis.html#cb112-11" tabindex="-1"></a><span class="fu">head</span>(grp_df) <span class="co"># col-1: sample; col-2: grouping info</span></span></code></pre></div>
<pre><code>## # A tibble: 6 × 2
##   sampleID        pathologic_M
##   &lt;chr&gt;           &lt;chr&gt;       
## 1 TCGA-05-4244-01 M1          
## 2 TCGA-05-4249-01 M0          
## 3 TCGA-05-4250-01 M0          
## 4 TCGA-05-4382-01 M0          
## 5 TCGA-05-4384-01 M0          
## 6 TCGA-05-4389-01 M0</code></pre>
<div class="sourceCode" id="cb114"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb114-1"><a href="molecular-data-analysis.html#cb114-1" tabindex="-1"></a>mol_dataset <span class="ot">&lt;-</span> <span class="st">&quot;TCGA.LUAD.sampleMap/HiSeqV2_percentile&quot;</span></span>
<span id="cb114-2"><a href="molecular-data-analysis.html#cb114-2" tabindex="-1"></a><span class="fu">vis_identifier_grp_comparison</span>(<span class="at">dataset =</span> mol_dataset, <span class="at">id =</span> <span class="st">&quot;TP53&quot;</span>, </span>
<span id="cb114-3"><a href="molecular-data-analysis.html#cb114-3" tabindex="-1"></a>                              <span class="at">grp_df =</span> grp_df)</span></code></pre></div>
<p><img src="_main_files/figure-html/vis-identifier-grp-comparison-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="correlation-analysis-3" class="section level3 hasAnchor" number="5.4.2">
<h3><span class="header-section-number">5.4.2</span> Correlation analysis<a href="molecular-data-analysis.html#correlation-analysis-3" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="vis_identifier_cor" class="section level4 hasAnchor" number="5.4.2.1">
<h4><span class="header-section-number">5.4.2.1</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_identifier_cor.html">vis_identifier_cor()</a><a href="molecular-data-analysis.html#vis_identifier_cor" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Calculate the correlation between two molecules value from genomics matrix UCSC Xena datasets.</p>
<ul>
<li>Basic use: <code>vis_identifier_cor(dataset= ,id1= ,dataset= ,id2= )</code></li>
</ul>
<div class="sourceCode" id="cb115"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb115-1"><a href="molecular-data-analysis.html#cb115-1" tabindex="-1"></a>dataset <span class="ot">&lt;-</span> <span class="st">&quot;TcgaTargetGtex_rsem_isoform_tpm&quot;</span></span>
<span id="cb115-2"><a href="molecular-data-analysis.html#cb115-2" tabindex="-1"></a><span class="fu">vis_identifier_cor</span>(<span class="at">dataset1 =</span> dataset, <span class="at">id1 =</span> <span class="st">&quot;TP53&quot;</span>,</span>
<span id="cb115-3"><a href="molecular-data-analysis.html#cb115-3" tabindex="-1"></a>                   <span class="at">dataset2 =</span> dataset, <span class="at">id2 =</span> <span class="st">&quot;KRAS&quot;</span>)</span></code></pre></div>
<pre><code>## Warning: package &#39;ggplot2&#39; was built under R version 4.2.3</code></pre>
<p><img src="_main_files/figure-html/vis-identifier-cor-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="vis_identifier_multi_cor" class="section level4 hasAnchor" number="5.4.2.2">
<h4><span class="header-section-number">5.4.2.2</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_identifier_multi_cor.html">vis_identifier_multi_cor()</a><a href="molecular-data-analysis.html#vis_identifier_multi_cor" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Calculate the pairwise correlation among multiple molecules value from one genomics matrix UCSC Xena dataset.</p>
<ul>
<li>Basic use: <code>vis_identifier_multi_cor(dataset= ,ids= )</code></li>
</ul>
<div class="sourceCode" id="cb117"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb117-1"><a href="molecular-data-analysis.html#cb117-1" tabindex="-1"></a>dataset <span class="ot">&lt;-</span> <span class="st">&quot;TcgaTargetGtex_rsem_isoform_tpm&quot;</span></span>
<span id="cb117-2"><a href="molecular-data-analysis.html#cb117-2" tabindex="-1"></a><span class="fu">vis_identifier_multi_cor</span>(<span class="at">dataset =</span> dataset,</span>
<span id="cb117-3"><a href="molecular-data-analysis.html#cb117-3" tabindex="-1"></a>                         <span class="at">ids =</span> <span class="fu">c</span>(<span class="st">&quot;TP53&quot;</span>, <span class="st">&quot;KRAS&quot;</span>, <span class="st">&quot;PTEN&quot;</span>))</span></code></pre></div>
<p><img src="_main_files/figure-html/vis-identifier-multi-cor-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="survival-analysis-2" class="section level3 hasAnchor" number="5.4.3">
<h3><span class="header-section-number">5.4.3</span> Survival analysis<a href="molecular-data-analysis.html#survival-analysis-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="vis_identifier_grp_surv" class="section level4 hasAnchor" number="5.4.3.1">
<h4><span class="header-section-number">5.4.3.1</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_identifier_grp_surv.html">vis_identifier_grp_surv()</a><a href="molecular-data-analysis.html#vis_identifier_grp_surv" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Perform the log-rank test of one molecule for one genomics matrix UCSC Xena dataset.</p>
<ul>
<li>Basic use: <code>vis_identifier_grp_surv(dataset= , id= , surv_df= )</code></li>
</ul>
<div class="sourceCode" id="cb118"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb118-1"><a href="molecular-data-analysis.html#cb118-1" tabindex="-1"></a><span class="co"># Firstly, prepare survival data of samples</span></span>
<span id="cb118-2"><a href="molecular-data-analysis.html#cb118-2" tabindex="-1"></a><span class="fu">library</span>(UCSCXenaTools)</span>
<span id="cb118-3"><a href="molecular-data-analysis.html#cb118-3" tabindex="-1"></a>cli_df <span class="ot">&lt;-</span> <span class="fu">XenaGenerate</span>(</span>
<span id="cb118-4"><a href="molecular-data-analysis.html#cb118-4" tabindex="-1"></a>  <span class="at">subset =</span> XenaDatasets <span class="sc">==</span> <span class="st">&quot;TCGA.LUAD.sampleMap/LUAD_clinicalMatrix&quot;</span></span>
<span id="cb118-5"><a href="molecular-data-analysis.html#cb118-5" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb118-6"><a href="molecular-data-analysis.html#cb118-6" tabindex="-1"></a>  <span class="fu">XenaQuery</span>() <span class="sc">%&gt;%</span></span>
<span id="cb118-7"><a href="molecular-data-analysis.html#cb118-7" tabindex="-1"></a>  <span class="fu">XenaDownload</span>() <span class="sc">%&gt;%</span></span>
<span id="cb118-8"><a href="molecular-data-analysis.html#cb118-8" tabindex="-1"></a>  <span class="fu">XenaPrepare</span>()</span>
<span id="cb118-9"><a href="molecular-data-analysis.html#cb118-9" tabindex="-1"></a>surv_df <span class="ot">&lt;-</span> cli_df[, <span class="fu">c</span>(<span class="st">&quot;sampleID&quot;</span>, <span class="st">&quot;days_to_death&quot;</span>, <span class="st">&quot;vital_status&quot;</span>)]</span>
<span id="cb118-10"><a href="molecular-data-analysis.html#cb118-10" tabindex="-1"></a>surv_df<span class="sc">$</span>vital_status <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(surv_df<span class="sc">$</span>vital_status <span class="sc">==</span> <span class="st">&quot;DECEASED&quot;</span>, <span class="dv">1</span>, <span class="dv">0</span>)</span>
<span id="cb118-11"><a href="molecular-data-analysis.html#cb118-11" tabindex="-1"></a>surv_df <span class="ot">=</span> <span class="fu">na.omit</span>(surv_df)</span>
<span id="cb118-12"><a href="molecular-data-analysis.html#cb118-12" tabindex="-1"></a><span class="fu">head</span>(surv_df)  <span class="co"># col-1: sample; col-2: survival time; col-3: survival status</span></span></code></pre></div>
<pre><code>## # A tibble: 6 × 3
##   sampleID        days_to_death vital_status
##   &lt;chr&gt;                   &lt;dbl&gt;        &lt;dbl&gt;
## 1 TCGA-05-4250-01           121            1
## 2 TCGA-05-4395-01             0            1
## 3 TCGA-05-4396-01           303            1
## 4 TCGA-05-4397-01           731            1
## 5 TCGA-05-4402-01           244            1
## 6 TCGA-05-4415-01            91            1</code></pre>
<div class="sourceCode" id="cb120"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb120-1"><a href="molecular-data-analysis.html#cb120-1" tabindex="-1"></a>mol_dataset <span class="ot">&lt;-</span> <span class="st">&quot;TCGA.LUAD.sampleMap/HiSeqV2_percentile&quot;</span></span>
<span id="cb120-2"><a href="molecular-data-analysis.html#cb120-2" tabindex="-1"></a><span class="fu">vis_identifier_grp_surv</span>(<span class="at">dataset =</span> mol_dataset, <span class="at">id =</span> <span class="st">&quot;KRAS&quot;</span>, </span>
<span id="cb120-3"><a href="molecular-data-analysis.html#cb120-3" tabindex="-1"></a>                        <span class="at">surv_df =</span> surv_df)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-identifier-grp-surv"></span>
<img src="_main_files/figure-html/vis-identifier-grp-surv-1.png" alt="The log-rank test (DSS) of mRNA KRAS for ne specific dataset" width="576" />
<p class="caption">
Figure 5.17: The log-rank test (DSS) of mRNA KRAS for ne specific dataset
</p>
</div>
<blockquote>
<p>By default, the best cutoff is decided. User can change it through the <code>cutoff_mode</code> parameter.</p>
</blockquote>
</div>
</div>
<div id="dimension-reduction-1" class="section level3 hasAnchor" number="5.4.4">
<h3><span class="header-section-number">5.4.4</span> Dimension reduction<a href="molecular-data-analysis.html#dimension-reduction-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="vis_identifier_dim_dist" class="section level4 hasAnchor" number="5.4.4.1">
<h4><span class="header-section-number">5.4.4.1</span> <a href="https://openbiox.github.io/UCSCXenaShiny/reference/vis_identifier_dim_dist.html">vis_identifier_dim_dist()</a><a href="molecular-data-analysis.html#vis_identifier_dim_dist" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Perform dimension reduction analysis of multiple molecules for samples in groups.</p>
<ul>
<li>Basic use: <code>vis_identifier_dim_dist(dataset= ,ids= , grp_df= )</code></li>
</ul>
<div class="sourceCode" id="cb121"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb121-1"><a href="molecular-data-analysis.html#cb121-1" tabindex="-1"></a><span class="co"># Firstly, prepare the grouping information of samples </span></span>
<span id="cb121-2"><a href="molecular-data-analysis.html#cb121-2" tabindex="-1"></a><span class="fu">library</span>(UCSCXenaTools)</span>
<span id="cb121-3"><a href="molecular-data-analysis.html#cb121-3" tabindex="-1"></a>cli_dataset <span class="ot">&lt;-</span> <span class="st">&quot;TCGA.LUAD.sampleMap/LUAD_clinicalMatrix&quot;</span></span>
<span id="cb121-4"><a href="molecular-data-analysis.html#cb121-4" tabindex="-1"></a>cli_df <span class="ot">&lt;-</span> <span class="fu">XenaGenerate</span>(</span>
<span id="cb121-5"><a href="molecular-data-analysis.html#cb121-5" tabindex="-1"></a>  <span class="at">subset =</span> XenaDatasets <span class="sc">==</span> cli_dataset</span>
<span id="cb121-6"><a href="molecular-data-analysis.html#cb121-6" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb121-7"><a href="molecular-data-analysis.html#cb121-7" tabindex="-1"></a>  <span class="fu">XenaQuery</span>() <span class="sc">%&gt;%</span></span>
<span id="cb121-8"><a href="molecular-data-analysis.html#cb121-8" tabindex="-1"></a>  <span class="fu">XenaDownload</span>() <span class="sc">%&gt;%</span></span>
<span id="cb121-9"><a href="molecular-data-analysis.html#cb121-9" tabindex="-1"></a>  <span class="fu">XenaPrepare</span>()</span>
<span id="cb121-10"><a href="molecular-data-analysis.html#cb121-10" tabindex="-1"></a>grp_df <span class="ot">=</span> cli_df[, <span class="fu">c</span>(<span class="st">&quot;sampleID&quot;</span>, <span class="st">&quot;gender&quot;</span>)]</span>
<span id="cb121-11"><a href="molecular-data-analysis.html#cb121-11" tabindex="-1"></a><span class="fu">head</span>(grp_df) <span class="co"># col-1: sample; col-2: grouping info</span></span></code></pre></div>
<pre><code>## # A tibble: 6 × 2
##   sampleID        gender
##   &lt;chr&gt;           &lt;chr&gt; 
## 1 TCGA-05-4244-01 MALE  
## 2 TCGA-05-4249-01 MALE  
## 3 TCGA-05-4250-01 FEMALE
## 4 TCGA-05-4382-01 MALE  
## 5 TCGA-05-4384-01 MALE  
## 6 TCGA-05-4389-01 MALE</code></pre>
<div class="sourceCode" id="cb123"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb123-1"><a href="molecular-data-analysis.html#cb123-1" tabindex="-1"></a>mol_dataset <span class="ot">&lt;-</span> <span class="st">&quot;TCGA.LUAD.sampleMap/HiSeqV2_percentile&quot;</span></span>
<span id="cb123-2"><a href="molecular-data-analysis.html#cb123-2" tabindex="-1"></a>ids <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;TP53&quot;</span>, <span class="st">&quot;KRAS&quot;</span>, <span class="st">&quot;PTEN&quot;</span>, <span class="st">&quot;MDM2&quot;</span>, <span class="st">&quot;CDKN1A&quot;</span>)</span>
<span id="cb123-3"><a href="molecular-data-analysis.html#cb123-3" tabindex="-1"></a><span class="fu">vis_identifier_dim_dist</span>(<span class="at">dataset =</span> mol_dataset,</span>
<span id="cb123-4"><a href="molecular-data-analysis.html#cb123-4" tabindex="-1"></a>                        <span class="at">ids =</span> ids, </span>
<span id="cb123-5"><a href="molecular-data-analysis.html#cb123-5" tabindex="-1"></a>                        <span class="at">grp_df =</span> grp_df)</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:vis-identifier-dim-dist"></span>
<img src="_main_files/figure-html/vis-identifier-dim-dist-1.png" alt="The dimension reduction analysis (PCA) of 5 mRNA molcules in BRCA cancer samples grouped by tissue codes" width="576" />
<p class="caption">
Figure 5.18: The dimension reduction analysis (PCA) of 5 mRNA molcules in BRCA cancer samples grouped by tissue codes
</p>
</div>

</div>
</div>
</div>
</div>



            </section>

          </div>
        </div>
      </div>
<a href="molecular-data-query.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="visit.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/06-pkg-4-analyze.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
