<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 12 Personalized TPC pipelines(2) | UCSCXenaShiny v2 Book</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 12 Personalized TPC pipelines(2) | UCSCXenaShiny v2 Book" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 12 Personalized TPC pipelines(2) | UCSCXenaShiny v2 Book" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Shensuo Li, Shixiang Wang" />


<meta name="date" content="2024-07-08" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="per-tpc-1.html"/>
<link rel="next" href="download.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#ucsc-xena-datasets"><i class="fa fa-check"></i><b>1.1</b> UCSC Xena Datasets</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#tcga-abbreviations"><i class="fa fa-check"></i><b>1.2</b> TCGA abbreviations</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#pcawg-abbreviations"><i class="fa fa-check"></i><b>1.3</b> PCAWG abbreviations</a></li>
</ul></li>
<li class="part"><span><b>I R package</b></span></li>
<li class="chapter" data-level="2" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i><b>2</b> Install</a>
<ul>
<li class="chapter" data-level="2.1" data-path="install.html"><a href="install.html#cran"><i class="fa fa-check"></i><b>2.1</b> CRAN</a></li>
<li class="chapter" data-level="2.2" data-path="install.html"><a href="install.html#github"><i class="fa fa-check"></i><b>2.2</b> Github</a></li>
<li class="chapter" data-level="2.3" data-path="install.html"><a href="install.html#conda"><i class="fa fa-check"></i><b>2.3</b> Conda</a></li>
<li class="chapter" data-level="2.4" data-path="install.html"><a href="install.html#docker"><i class="fa fa-check"></i><b>2.4</b> Docker</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="build-in.html"><a href="build-in.html"><i class="fa fa-check"></i><b>3</b> Built-in datasets</a>
<ul>
<li class="chapter" data-level="3.1" data-path="build-in.html"><a href="build-in.html#tcga"><i class="fa fa-check"></i><b>3.1</b> TCGA</a></li>
<li class="chapter" data-level="3.2" data-path="build-in.html"><a href="build-in.html#pcawg"><i class="fa fa-check"></i><b>3.2</b> PCAWG</a></li>
<li class="chapter" data-level="3.3" data-path="build-in.html"><a href="build-in.html#ccle"><i class="fa fa-check"></i><b>3.3</b> CCLE</a></li>
<li class="chapter" data-level="3.4" data-path="build-in.html"><a href="build-in.html#zenodo"><i class="fa fa-check"></i><b>3.4</b> Zenodo</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="build-in.html"><a href="build-in.html#tumor-infiltration-estimations"><i class="fa fa-check"></i><b>3.4.1</b> tumor infiltration estimations</a></li>
<li class="chapter" data-level="3.4.2" data-path="build-in.html"><a href="build-in.html#ssgsea-pathway-activities"><i class="fa fa-check"></i><b>3.4.2</b> ssGSEA pathway activities</a></li>
<li class="chapter" data-level="3.4.3" data-path="build-in.html"><a href="build-in.html#other-tcga-annotations"><i class="fa fa-check"></i><b>3.4.3</b> other TCGA annotations</a></li>
<li class="chapter" data-level="3.4.4" data-path="build-in.html"><a href="build-in.html#identifier-repository"><i class="fa fa-check"></i><b>3.4.4</b> identifier repository</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-query.html"><a href="data-query.html"><i class="fa fa-check"></i><b>4</b> Molecular data query</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-query.html"><a href="data-query.html#tcga-query"><i class="fa fa-check"></i><b>4.1</b> TCGA query</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="data-query.html"><a href="data-query.html#get_pancan_gene_value"><i class="fa fa-check"></i><b>4.1.1</b> get_pancan_gene_value()</a></li>
<li class="chapter" data-level="4.1.2" data-path="data-query.html"><a href="data-query.html#get_pancan_transcript_value"><i class="fa fa-check"></i><b>4.1.2</b> get_pancan_transcript_value()</a></li>
<li class="chapter" data-level="4.1.3" data-path="data-query.html"><a href="data-query.html#get_pancan_protein_value"><i class="fa fa-check"></i><b>4.1.3</b> get_pancan_protein_value()</a></li>
<li class="chapter" data-level="4.1.4" data-path="data-query.html"><a href="data-query.html#get_pancan_mutation_status"><i class="fa fa-check"></i><b>4.1.4</b> get_pancan_mutation_status()</a></li>
<li class="chapter" data-level="4.1.5" data-path="data-query.html"><a href="data-query.html#get_pancan_cn_value"><i class="fa fa-check"></i><b>4.1.5</b> get_pancan_cn_value()</a></li>
<li class="chapter" data-level="4.1.6" data-path="data-query.html"><a href="data-query.html#get_pancan_methylation_value"><i class="fa fa-check"></i><b>4.1.6</b> get_pancan_methylation_value()</a></li>
<li class="chapter" data-level="4.1.7" data-path="data-query.html"><a href="data-query.html#get_pancan_mirna_value"><i class="fa fa-check"></i><b>4.1.7</b> get_pancan_miRNA_value()</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="data-query.html"><a href="data-query.html#pcawg-query"><i class="fa fa-check"></i><b>4.2</b> PCAWG query</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="data-query.html"><a href="data-query.html#get_pcawg_gene_value"><i class="fa fa-check"></i><b>4.2.1</b> get_pcawg_gene_value()</a></li>
<li class="chapter" data-level="4.2.2" data-path="data-query.html"><a href="data-query.html#get_pcawg_fusion_value"><i class="fa fa-check"></i><b>4.2.2</b> get_pcawg_fusion_value()</a></li>
<li class="chapter" data-level="4.2.3" data-path="data-query.html"><a href="data-query.html#get_pcawg_mirna_value"><i class="fa fa-check"></i><b>4.2.3</b> get_pcawg_miRNA_value()</a></li>
<li class="chapter" data-level="4.2.4" data-path="data-query.html"><a href="data-query.html#get_pcawg_promoter_value"><i class="fa fa-check"></i><b>4.2.4</b> get_pcawg_promoter_value()</a></li>
<li class="chapter" data-level="4.2.5" data-path="data-query.html"><a href="data-query.html#get_pcawg_apobec_mutagenesis_value"><i class="fa fa-check"></i><b>4.2.5</b> get_pcawg_APOBEC_mutagenesis_value()</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="data-query.html"><a href="data-query.html#ccle-query"><i class="fa fa-check"></i><b>4.3</b> CCLE query</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="data-query.html"><a href="data-query.html#get_ccle_gene_value"><i class="fa fa-check"></i><b>4.3.1</b> get_ccle_gene_value()</a></li>
<li class="chapter" data-level="4.3.2" data-path="data-query.html"><a href="data-query.html#get_ccle_protein_value"><i class="fa fa-check"></i><b>4.3.2</b> get_ccle_protein_value()</a></li>
<li class="chapter" data-level="4.3.3" data-path="data-query.html"><a href="data-query.html#get_ccle_mutation_status"><i class="fa fa-check"></i><b>4.3.3</b> get_ccle_mutation_status()</a></li>
<li class="chapter" data-level="4.3.4" data-path="data-query.html"><a href="data-query.html#get_ccle_cn_value"><i class="fa fa-check"></i><b>4.3.4</b> get_ccle_cn_value()</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="data-query.html"><a href="data-query.html#general-query"><i class="fa fa-check"></i><b>4.4</b> General query</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="data-query.html"><a href="data-query.html#query_pancan_value"><i class="fa fa-check"></i><b>4.4.1</b> query_pancan_value()</a></li>
<li class="chapter" data-level="4.4.2" data-path="data-query.html"><a href="data-query.html#query_molecule_value"><i class="fa fa-check"></i><b>4.4.2</b> query_molecule_value()</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-analysis.html"><a href="data-analysis.html"><i class="fa fa-check"></i><b>5</b> Molecular data analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-analysis.html"><a href="data-analysis.html#tcga-analysis"><i class="fa fa-check"></i><b>5.1</b> TCGA analysis</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="data-analysis.html"><a href="data-analysis.html#comparison-analysis"><i class="fa fa-check"></i><b>5.1.1</b> Comparison analysis</a></li>
<li class="chapter" data-level="5.1.2" data-path="data-analysis.html"><a href="data-analysis.html#correlation-analysis"><i class="fa fa-check"></i><b>5.1.2</b> Correlation analysis</a></li>
<li class="chapter" data-level="5.1.3" data-path="data-analysis.html"><a href="data-analysis.html#survival-analysis"><i class="fa fa-check"></i><b>5.1.3</b> Survival analysis</a></li>
<li class="chapter" data-level="5.1.4" data-path="data-analysis.html"><a href="data-analysis.html#dimension-reduction"><i class="fa fa-check"></i><b>5.1.4</b> Dimension reduction</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="data-analysis.html"><a href="data-analysis.html#pcawg-analysis"><i class="fa fa-check"></i><b>5.2</b> PCAWG analysis</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="data-analysis.html"><a href="data-analysis.html#comparsion-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Comparsion analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="data-analysis.html"><a href="data-analysis.html#correlation-analysis-1"><i class="fa fa-check"></i><b>5.2.2</b> Correlation analysis</a></li>
<li class="chapter" data-level="5.2.3" data-path="data-analysis.html"><a href="data-analysis.html#survival-analysis-1"><i class="fa fa-check"></i><b>5.2.3</b> Survival analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="data-analysis.html"><a href="data-analysis.html#ccle-analysis"><i class="fa fa-check"></i><b>5.3</b> CCLE analysis</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-analysis.html"><a href="data-analysis.html#comparsion-analysis-1"><i class="fa fa-check"></i><b>5.3.1</b> Comparsion analysis</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-analysis.html"><a href="data-analysis.html#correlation-analysis-2"><i class="fa fa-check"></i><b>5.3.2</b> Correlation analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="data-analysis.html"><a href="data-analysis.html#general-analysis"><i class="fa fa-check"></i><b>5.4</b> General analysis</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="data-analysis.html"><a href="data-analysis.html#comparison-analysis-1"><i class="fa fa-check"></i><b>5.4.1</b> Comparison analysis</a></li>
<li class="chapter" data-level="5.4.2" data-path="data-analysis.html"><a href="data-analysis.html#correlation-analysis-3"><i class="fa fa-check"></i><b>5.4.2</b> Correlation analysis</a></li>
<li class="chapter" data-level="5.4.3" data-path="data-analysis.html"><a href="data-analysis.html#survival-analysis-2"><i class="fa fa-check"></i><b>5.4.3</b> Survival analysis</a></li>
<li class="chapter" data-level="5.4.4" data-path="data-analysis.html"><a href="data-analysis.html#dimension-reduction-1"><i class="fa fa-check"></i><b>5.4.4</b> Dimension reduction</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Shiny App</b></span></li>
<li class="chapter" data-level="6" data-path="visit.html"><a href="visit.html"><i class="fa fa-check"></i><b>6</b> Visit</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visit.html"><a href="visit.html#server-deployments"><i class="fa fa-check"></i><b>6.1</b> Server deployments</a></li>
<li class="chapter" data-level="6.2" data-path="visit.html"><a href="visit.html#local-deployment"><i class="fa fa-check"></i><b>6.2</b> Local deployment</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="home.html"><a href="home.html"><i class="fa fa-check"></i><b>7</b> Homepage</a>
<ul>
<li class="chapter" data-level="7.1" data-path="home.html"><a href="home.html#quick-molecule-exploration"><i class="fa fa-check"></i><b>7.1</b> Quick Molecule Exploration</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="home.html"><a href="home.html#daily-gene"><i class="fa fa-check"></i><b>7.1.1</b> Daily Gene</a></li>
<li class="chapter" data-level="7.1.2" data-path="home.html"><a href="home.html#pan-cancer-query"><i class="fa fa-check"></i><b>7.1.2</b> Pan-cancer Query</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="home.html"><a href="home.html#slick-gallery"><i class="fa fa-check"></i><b>7.2</b> Slick Gallery</a></li>
<li class="chapter" data-level="7.3" data-path="home.html"><a href="home.html#tcga-analysis-guide"><i class="fa fa-check"></i><b>7.3</b> TCGA Analysis Guide</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="repo.html"><a href="repo.html"><i class="fa fa-check"></i><b>8</b> Repository</a>
<ul>
<li class="chapter" data-level="8.1" data-path="repo.html"><a href="repo.html#show-metadata"><i class="fa fa-check"></i><b>8.1</b> Show Metadata</a></li>
<li class="chapter" data-level="8.2" data-path="repo.html"><a href="repo.html#request-data"><i class="fa fa-check"></i><b>8.2</b> Request Data</a></li>
<li class="chapter" data-level="8.3" data-path="repo.html"><a href="repo.html#analyze-data"><i class="fa fa-check"></i><b>8.3</b> Analyze Data</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="general-da.html"><a href="general-da.html"><i class="fa fa-check"></i><b>9</b> General Dataset Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="general-da.html"><a href="general-da.html#m-vis-identifier-cor"><i class="fa fa-check"></i><b>9.1</b> Scatter-Correlation</a></li>
<li class="chapter" data-level="9.2" data-path="general-da.html"><a href="general-da.html#m-vis-identifier-multi-cor"><i class="fa fa-check"></i><b>9.2</b> Matrix-Correlation</a></li>
<li class="chapter" data-level="9.3" data-path="general-da.html"><a href="general-da.html#m-vis-identifier-grp-comparison"><i class="fa fa-check"></i><b>9.3</b> Group-Comparison</a></li>
<li class="chapter" data-level="9.4" data-path="general-da.html"><a href="general-da.html#m-vis-identifier-grp-surv"><i class="fa fa-check"></i><b>9.4</b> Survival-Analysis</a></li>
<li class="chapter" data-level="9.5" data-path="general-da.html"><a href="general-da.html#m-vis-identifier-dim-dist"><i class="fa fa-check"></i><b>9.5</b> Dimension-Distribution</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="custom-tpc.html"><a href="custom-tpc.html"><i class="fa fa-check"></i><b>10</b> Custom TPC modules</a>
<ul>
<li class="chapter" data-level="10.1" data-path="custom-tpc.html"><a href="custom-tpc.html#tcga-1"><i class="fa fa-check"></i><b>10.1</b> TCGA</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="custom-tpc.html"><a href="custom-tpc.html#comparison"><i class="fa fa-check"></i><b>10.1.1</b> Comparison</a></li>
<li class="chapter" data-level="10.1.2" data-path="custom-tpc.html"><a href="custom-tpc.html#correlation"><i class="fa fa-check"></i><b>10.1.2</b> Correlation</a></li>
<li class="chapter" data-level="10.1.3" data-path="custom-tpc.html"><a href="custom-tpc.html#survival-analysis-3"><i class="fa fa-check"></i><b>10.1.3</b> Survival analysis</a></li>
<li class="chapter" data-level="10.1.4" data-path="custom-tpc.html"><a href="custom-tpc.html#dimension-reduction-2"><i class="fa fa-check"></i><b>10.1.4</b> Dimension Reduction</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="custom-tpc.html"><a href="custom-tpc.html#pcawg-1"><i class="fa fa-check"></i><b>10.2</b> PCAWG</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="custom-tpc.html"><a href="custom-tpc.html#comparison-analysis-2"><i class="fa fa-check"></i><b>10.2.1</b> Comparison analysis</a></li>
<li class="chapter" data-level="10.2.2" data-path="custom-tpc.html"><a href="custom-tpc.html#correlation-analysis-4"><i class="fa fa-check"></i><b>10.2.2</b> Correlation analysis</a></li>
<li class="chapter" data-level="10.2.3" data-path="custom-tpc.html"><a href="custom-tpc.html#survival-analysis-4"><i class="fa fa-check"></i><b>10.2.3</b> Survival analysis</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="custom-tpc.html"><a href="custom-tpc.html#ccle-1"><i class="fa fa-check"></i><b>10.3</b> CCLE</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="custom-tpc.html"><a href="custom-tpc.html#comparison-analysis-3"><i class="fa fa-check"></i><b>10.3.1</b> Comparison analysis</a></li>
<li class="chapter" data-level="10.3.2" data-path="custom-tpc.html"><a href="custom-tpc.html#correlation-analysis-5"><i class="fa fa-check"></i><b>10.3.2</b> Correlation analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="per-tpc-1.html"><a href="per-tpc-1.html"><i class="fa fa-check"></i><b>11</b> Personalized TPC pipelines(1)</a>
<ul>
<li class="chapter" data-level="11.1" data-path="per-tpc-1.html"><a href="per-tpc-1.html#extensive-tumor-data"><i class="fa fa-check"></i><b>11.1</b> Extensive tumor data</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="per-tpc-1.html"><a href="per-tpc-1.html#compiling-types-of-identifiers"><i class="fa fa-check"></i><b>11.1.1</b> Compiling types of identifiers</a></li>
<li class="chapter" data-level="11.1.2" data-path="per-tpc-1.html"><a href="per-tpc-1.html#reference-of-full-identifiers"><i class="fa fa-check"></i><b>11.1.2</b> Reference of full identifiers</a></li>
<li class="chapter" data-level="11.1.3" data-path="per-tpc-1.html"><a href="per-tpc-1.html#user-defined-identifiers"><i class="fa fa-check"></i><b>11.1.3</b> User-defined identifiers</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="per-tpc-1.html"><a href="per-tpc-1.html#sample-processing"><i class="fa fa-check"></i><b>11.2</b> Sample processing</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="per-tpc-1.html"><a href="per-tpc-1.html#sample-filtering"><i class="fa fa-check"></i><b>11.2.1</b> Sample filtering</a></li>
<li class="chapter" data-level="11.2.2" data-path="per-tpc-1.html"><a href="per-tpc-1.html#sample-grouping"><i class="fa fa-check"></i><b>11.2.2</b> Sample grouping</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="per-tpc-1.html"><a href="per-tpc-1.html#data-fetch"><i class="fa fa-check"></i><b>11.3</b> Data fetch</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="per-tpc-1.html"><a href="per-tpc-1.html#single-data"><i class="fa fa-check"></i><b>11.3.1</b> Single data</a></li>
<li class="chapter" data-level="11.3.2" data-path="per-tpc-1.html"><a href="per-tpc-1.html#multiple-data"><i class="fa fa-check"></i><b>11.3.2</b> Multiple data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="per-tpc-2.html"><a href="per-tpc-2.html"><i class="fa fa-check"></i><b>12</b> Personalized TPC pipelines(2)</a>
<ul>
<li class="chapter" data-level="12.1" data-path="per-tpc-2.html"><a href="per-tpc-2.html#general-steps"><i class="fa fa-check"></i><b>12.1</b> General steps</a></li>
<li class="chapter" data-level="12.2" data-path="per-tpc-2.html"><a href="per-tpc-2.html#correlation-analysis-6"><i class="fa fa-check"></i><b>12.2</b> Correlation analysis</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="per-tpc-2.html"><a href="per-tpc-2.html#individual-mode"><i class="fa fa-check"></i><b>12.2.1</b> Individual Mode</a></li>
<li class="chapter" data-level="12.2.2" data-path="per-tpc-2.html"><a href="per-tpc-2.html#pan-cancer-mode"><i class="fa fa-check"></i><b>12.2.2</b> Pan-cancer Mode</a></li>
<li class="chapter" data-level="12.2.3" data-path="per-tpc-2.html"><a href="per-tpc-2.html#batch-screen-mode"><i class="fa fa-check"></i><b>12.2.3</b> Batch Screen Mode</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="per-tpc-2.html"><a href="per-tpc-2.html#comparison-analysis-4"><i class="fa fa-check"></i><b>12.3</b> Comparison analysis</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="per-tpc-2.html"><a href="per-tpc-2.html#individual-mode-1"><i class="fa fa-check"></i><b>12.3.1</b> Individual Mode</a></li>
<li class="chapter" data-level="12.3.2" data-path="per-tpc-2.html"><a href="per-tpc-2.html#pan-cancer-mode-1"><i class="fa fa-check"></i><b>12.3.2</b> Pan-cancer Mode</a></li>
<li class="chapter" data-level="12.3.3" data-path="per-tpc-2.html"><a href="per-tpc-2.html#batch-screen-mode-1"><i class="fa fa-check"></i><b>12.3.3</b> Batch Screen Mode</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="per-tpc-2.html"><a href="per-tpc-2.html#survival-analysis-5"><i class="fa fa-check"></i><b>12.4</b> Survival analysis</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="per-tpc-2.html"><a href="per-tpc-2.html#individual-mode-2"><i class="fa fa-check"></i><b>12.4.1</b> Individual Mode</a></li>
<li class="chapter" data-level="12.4.2" data-path="per-tpc-2.html"><a href="per-tpc-2.html#pan-cancer-mode-2"><i class="fa fa-check"></i><b>12.4.2</b> Pan-cancer Mode</a></li>
<li class="chapter" data-level="12.4.3" data-path="per-tpc-2.html"><a href="per-tpc-2.html#batch-screen-mode-2"><i class="fa fa-check"></i><b>12.4.3</b> Batch Screen Mode</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="per-tpc-2.html"><a href="per-tpc-2.html#cross-omics-analysis"><i class="fa fa-check"></i><b>12.5</b> Cross-Omics analysis</a>
<ul>
<li class="chapter" data-level="12.5.1" data-path="per-tpc-2.html"><a href="per-tpc-2.html#gene-cross-omics-analysis"><i class="fa fa-check"></i><b>12.5.1</b> Gene Cross-Omics Analysis</a></li>
<li class="chapter" data-level="12.5.2" data-path="per-tpc-2.html"><a href="per-tpc-2.html#pathway-cross-omics-analysis"><i class="fa fa-check"></i><b>12.5.2</b> Pathway Cross-Omics Analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="download.html"><a href="download.html"><i class="fa fa-check"></i><b>13</b> Download modules</a>
<ul>
<li class="chapter" data-level="13.1" data-path="download.html"><a href="download.html#download-tpc-datasets"><i class="fa fa-check"></i><b>13.1</b> Download TPC datasets</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="download.html"><a href="download.html#omics-molecular-data"><i class="fa fa-check"></i><b>13.1.1</b> Omics molecular data</a></li>
<li class="chapter" data-level="13.1.2" data-path="download.html"><a href="download.html#non-omics-annotation-data"><i class="fa fa-check"></i><b>13.1.2</b> Non-omics annotation data</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="download.html"><a href="download.html#download-ucsc-xena-datasets"><i class="fa fa-check"></i><b>13.2</b> Download UCSC Xena datasets</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/openbiox/UCSCXenaShiny" target="blank">UCSCXenaShiny Github</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">UCSCXenaShiny v2 Book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="per-tpc-2" class="section level1 hasAnchor" number="12">
<h1><span class="header-section-number">Chapter 12</span> Personalized TPC pipelines(2)<a href="per-tpc-2.html#per-tpc-2" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="general-steps" class="section level2 hasAnchor" number="12.1">
<h2><span class="header-section-number">12.1</span> General steps<a href="per-tpc-2.html#general-steps" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><strong>S1: Preset</strong> — Set personalized samples and tumor data with following general steps.</p>
<ul>
<li>“<strong>Modify datasets</strong>”: View the dataset sources for each molecular type. When there are alternative datasets for the same molecular type, users can switch a suitable one based on purpose (referring to Chapter <a href="custom-tpc.html#custom-tpc">10</a>).</li>
<li>“<strong>Choose cancer</strong>”: Choose one cancer type for downstream association analysis.</li>
<li>“<strong>Filter samples</strong>”: Upon the selection of one tumor, further refine the samples based on tissue codes or personalized filtering module.</li>
<li>“<strong>Upload metadata</strong>”: Upload user-defined tumor data for joint analysis.</li>
<li>“<strong>Add signature</strong>”: Design custom molecular signature for joint analysis.</li>
</ul>
<p><strong>S2: Get data</strong> — Select and fetch identifier values; set grouping information.</p>
<p><strong>S3: Analyze &amp; Visualize</strong> — Modify diverse parameters; perform analysis; download results.</p>
<ul>
<li>“<strong>Set analyzation parameters</strong>”: Select different statistical methods</li>
<li>“<strong>Set visualization parameters</strong>”: Adjust plot color, title text, etc.</li>
<li>“<strong>Download results</strong>”: Download three types of results, including raw data, detailed statistical results and plot results. (Note: No plot for batch-screen mode).</li>
</ul>
<blockquote>
<p>Notes: We will introduce various analysis scenarios of TPC using TCGA as example.</p>
</blockquote>
</div>
<div id="correlation-analysis-6" class="section level2 hasAnchor" number="12.2">
<h2><span class="header-section-number">12.2</span> Correlation analysis<a href="per-tpc-2.html#correlation-analysis-6" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1301"></span>
<img src="images/p1301.png" alt="3 modes of correlation analysis" width="100%" />
<p class="caption">
Figure 12.1: 3 modes of correlation analysis
</p>
</div>
<div id="individual-mode" class="section level3 hasAnchor" number="12.2.1">
<h3><span class="header-section-number">12.2.1</span> Individual Mode<a href="per-tpc-2.html#individual-mode" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Perform the correlation analysis between any <strong>two identifiers</strong> of samples from <strong>one tumor</strong>.</p>
<p>The following is a simple example. (The steps not mentioned in above figure imply default operations.)</p>
<p><strong>S1: Preset</strong></p>
<ul>
<li>“<strong>S1.2 Choose cancer</strong>”: choose the <strong><em>CHOL</em></strong> (Cholangiocarcinoma) cancer type</li>
<li>“<strong>S1.3 Filter samples</strong>”: choose the <strong><em>primary tumor</em></strong> samples via the quick filter.</li>
</ul>
<p><strong>S2: Get data</strong></p>
<ul>
<li>“<strong>S2.1 Get data for X-axis</strong>”: Molecular profile→mRNA Expression→<strong><em>TP53</em></strong></li>
<li>“<strong>S2.2 Get data for Y-axis</strong>”: Infiltration→CIBERSORT→ <strong><em>Monocyte</em></strong></li>
</ul>
<p><strong>S3: Analyze &amp; Visualize</strong></p>
<ul>
<li>“<strong>S3.1 Set analysis parameters</strong>”: <strong><em>Spearman</em></strong></li>
<li>“<strong>S3.2 Set visualization parameters</strong>”: Adjust the main title name.</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1302"></span>
<img src="images/p1302.png" alt="Example of correlation analysis in individual mode" width="100%" />
<p class="caption">
Figure 12.2: Example of correlation analysis in individual mode
</p>
</div>
</div>
<div id="pan-cancer-mode" class="section level3 hasAnchor" number="12.2.2">
<h3><span class="header-section-number">12.2.2</span> Pan-cancer Mode<a href="per-tpc-2.html#pan-cancer-mode" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Perform the correlation analysis between any <strong>two identifiers</strong> of samples across <strong>multiple tumors</strong>.</p>
<p>The following is a simple example. (The steps not mentioned in above figure imply default operations.)</p>
<p><strong>S1: Preset</strong></p>
<ul>
<li>“<strong>S1.1 Modify datasets</strong>”: Select the <strong>Norm_Count</strong> dataset for mRNA Expression dataset.</li>
<li>“<strong>S1.2 Choose cancer</strong>”: choose all TCGA cancer types.</li>
<li>“<strong>S1.3 Filter samples</strong>”: choose the <strong><em>non-normal</em></strong> samples via the quick filter.</li>
</ul>
<p><strong>S2: Get data</strong></p>
<ul>
<li>“<strong>S2.1 Get data for X-axis</strong>”: Molecular profile→mRNA Expression→<strong><em>HDAC1</em></strong></li>
<li>“<strong>S2.2 Get data for Y-axis</strong>”: Molecular profile→mRNA Expression→<strong><em>APOE</em></strong></li>
</ul>
<p><strong>S3: Analyze &amp; Visualize</strong></p>
<ul>
<li>“<strong>S3.1 Set analysis parameters</strong>”: <strong><em>Pearson</em></strong></li>
<li>“<strong>S3.2 Set visualization parameters</strong>”: Adjust the names of main title and axis title.</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1303"></span>
<img src="images/p1303.png" alt="Example of correlation analysis in pan-cancer mode" width="100%" />
<p class="caption">
Figure 12.3: Example of correlation analysis in pan-cancer mode
</p>
</div>
</div>
<div id="batch-screen-mode" class="section level3 hasAnchor" number="12.2.3">
<h3><span class="header-section-number">12.2.3</span> Batch Screen Mode<a href="per-tpc-2.html#batch-screen-mode" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Perform the correlation analysis between <strong>multiple identifiers</strong> and <strong>one identifier</strong> of samples from <strong>one tumor</strong>.</p>
<p>The following is a simple example. (The steps not mentioned in above figure imply default operations.)</p>
<p><strong>S1: Preset</strong></p>
<ul>
<li>“<strong>S1.2 Choose cancer</strong>”: choose the <strong><em>CHOL</em></strong> (Cholangiocarcinoma) cancer type</li>
<li>“<strong>S1.3 Filter samples</strong>”: choose the <strong><em>primary tumor</em></strong> samples via the quick filter.</li>
</ul>
<p><strong>S2: Get data</strong></p>
<ul>
<li>“<strong>S2.1 Get batch data for X-axis</strong>”: Molecular profile→mRNA Expression→ genes of HALLMARK_APOPTOSIS pathways</li>
<li>“<strong>S2.2 Get data for Y-axis</strong>”: Immune Infiltration→CIBERSORT→ <strong><em>Monocyte</em></strong></li>
</ul>
<p><strong>S3: Analyze</strong></p>
<ul>
<li>“<strong>S3.1 Set analysis parameters</strong>”: <strong><em>Spearman</em></strong></li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1304"></span>
<img src="images/p1304.png" alt="Example of correlation analysis in batch screen mode" width="100%" />
<p class="caption">
Figure 12.4: Example of correlation analysis in batch screen mode
</p>
</div>
</div>
</div>
<div id="comparison-analysis-4" class="section level2 hasAnchor" number="12.3">
<h2><span class="header-section-number">12.3</span> Comparison analysis<a href="per-tpc-2.html#comparison-analysis-4" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1305"></span>
<img src="images/p1305.png" alt="3 modes of comparison analysis" width="100%" />
<p class="caption">
Figure 12.5: 3 modes of comparison analysis
</p>
</div>
<div id="individual-mode-1" class="section level3 hasAnchor" number="12.3.1">
<h3><span class="header-section-number">12.3.1</span> Individual Mode<a href="per-tpc-2.html#individual-mode-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Perform the comparison analysis of <strong>one identifier</strong> between two groups of samples from <strong>one tumor</strong>.</p>
<p>The following is a simple example. (The steps not mentioned in above figure imply default operations.)</p>
<p><strong>S1: Preset</strong></p>
<ul>
<li>“<strong>S1.2 Choose cancer</strong>”: choose the <strong><em>BRCA</em></strong> (Breast invasive carcinoma) cancer type</li>
<li>“<strong>S1.3 Filter samples</strong>”: choose the <strong><em>non-normal</em></strong> samples via the quick filter.</li>
</ul>
<p><strong>S2: Get data</strong></p>
<ul>
<li>“<strong>S2.1 Divide 2 groups by one condition</strong>”: Group1→<strong>Primary</strong> tumor samples; Group2→<strong>Metastatic</strong> tumor samples</li>
<li>“<strong>S2.2 Get data for comparison</strong>”: Molecular profile→mRNA Expression→<strong><em>POSTN</em></strong></li>
</ul>
<p><strong>S3: Analyze &amp; Visualize</strong></p>
<ul>
<li>“<strong>S3.1 Set analysis parameters</strong>”: <strong><em>Wilcoxon test</em></strong></li>
<li>“<strong>S3.2 Set visualization parameters</strong>”: Remove x-axis title.</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1306"></span>
<img src="images/p1306.png" alt="Example of comparison analysis in individual mode" width="100%" />
<p class="caption">
Figure 12.6: Example of comparison analysis in individual mode
</p>
</div>
</div>
<div id="pan-cancer-mode-1" class="section level3 hasAnchor" number="12.3.2">
<h3><span class="header-section-number">12.3.2</span> Pan-cancer Mode<a href="per-tpc-2.html#pan-cancer-mode-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Perform the comparison analysis of <strong>one identifier</strong> between two groups of samples across <strong>multiple tumors</strong>.</p>
<p>The following is a simple example. (The steps not mentioned in above figure imply default operations.)</p>
<p><strong>S1: Preset</strong></p>
<ul>
<li>“<strong>S1.2 Choose cancer</strong>”: choose the all TCGA cancer types.</li>
<li>“<strong>S1.3 Filter samples</strong>”: choose the <strong><em>primary tumor</em></strong> samples via the quick filter. Further choose samples with age above 60 via the exact filter.</li>
</ul>
<p><strong>S2: Get data</strong></p>
<ul>
<li>“<strong>S2.1 Divide 2 groups by one condition</strong>”: Group1→ gene PTEN mutation; Group2→ gene PTEN wild-type.</li>
<li>“<strong>S2.2 Get data for comparison</strong>”: Tumor index → Tumor Stemness →<strong><em>RNAss</em></strong></li>
</ul>
<p><strong>S3: Analyze &amp; Visualize</strong></p>
<ul>
<li>“<strong>S3.1 Set analysis parameters</strong>”: <strong><em>t-test</em></strong></li>
<li>“<strong>S3.2 Set visualization parameters</strong>”: adjust x-axis title.</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1307"></span>
<img src="images/p1307.png" alt="Example of comparison analysis in pan-cancer mode" width="100%" />
<p class="caption">
Figure 12.7: Example of comparison analysis in pan-cancer mode
</p>
</div>
</div>
<div id="batch-screen-mode-1" class="section level3 hasAnchor" number="12.3.3">
<h3><span class="header-section-number">12.3.3</span> Batch Screen Mode<a href="per-tpc-2.html#batch-screen-mode-1" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Perform the comparison analysis of <strong>multiple identifier</strong> between two groups of samples from <strong>one tumor</strong>.</p>
<p>The following is a simple example. (The steps not mentioned in above figure imply default operations.)</p>
<p><strong>S1: Preset</strong></p>
<ul>
<li>“<strong>S1.2 Choose cancer</strong>”: choose the <strong><em>BRCA</em></strong> (Breast invasive carcinoma) cancer type</li>
<li>“<strong>S1.3 Filter samples</strong>”: choose the <strong><em>non-normal</em></strong> samples via the quick filter.</li>
</ul>
<p><strong>S2: Get data</strong></p>
<ul>
<li>“<strong>S2.1 Divide 2 groups by one condition</strong>”: Group1→<strong>Primary</strong> tumor samples; Group2→<strong>Metastatic</strong> tumor samples</li>
<li>“<strong>S2.2 Get data for comparison</strong>”: Immune Infiltration→XCELL→ <strong><em>Monocyte</em></strong></li>
</ul>
<p><strong>S3: Analyze</strong></p>
<ul>
<li>“<strong>S3.1 Set analysis parameters</strong>”: <strong><em>Wilcoxon test</em></strong></li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1308"></span>
<img src="images/p1308.png" alt="Example of comparison analysis in batch screen mode" width="100%" />
<p class="caption">
Figure 12.8: Example of comparison analysis in batch screen mode
</p>
</div>
</div>
</div>
<div id="survival-analysis-5" class="section level2 hasAnchor" number="12.4">
<h2><span class="header-section-number">12.4</span> Survival analysis<a href="per-tpc-2.html#survival-analysis-5" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>By default, both log-rank and cox regression survival analysis are performed for two groups of samples divided according to one condition. Here, we devised a switch “<strong><em>Whether use initial data before grouping?</em></strong>” at <strong>S3.1</strong> steps with following function:</p>
<ul>
<li><p>If its status is off , the above default analysis will performed;</p></li>
<li><p>If its status is on and the grouping condition is <strong>continuous</strong>:</p>
<ul>
<li>For log-rank test, it will search the <strong>optimal cutoff</strong> with most significant survival difference;</li>
<li>For Cox regression, it will directly build the cox model based on the continuous variable.</li>
</ul></li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1309"></span>
<img src="images/p1309.png" alt="3 modes of survival analysis" width="100%" />
<p class="caption">
Figure 12.9: 3 modes of survival analysis
</p>
</div>
<div id="individual-mode-2" class="section level3 hasAnchor" number="12.4.1">
<h3><span class="header-section-number">12.4.1</span> Individual Mode<a href="per-tpc-2.html#individual-mode-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Perform the survival analysis between two groups of samples generated by <strong>one identifier</strong> from <strong>one tumor</strong>.</p>
<p>The following is a simple example. (The steps not mentioned in above figure imply default operations.)</p>
<p><strong>S1: Preset</strong></p>
<ul>
<li>“<strong>S1.1 Modify datasets</strong>”: Select the 27K Methylation datasets; Select the cg15206330 site to represent TP53 methylation</li>
<li>“<strong>S1.2 Choose cancer</strong>”: choose the <strong><em>BRCA</em></strong> (Breast invasive carcinoma) cancer type</li>
<li>“<strong>S1.3 Filter samples</strong>”: choose the <strong><em>non-normal</em></strong> samples via the quick filter.</li>
</ul>
<p><strong>S2: Get data</strong></p>
<ul>
<li>“<strong>S2.1 Select survival endpoint</strong>”: Select the OS event</li>
<li>“<strong>S2.2 Divide 2 groups by one condition</strong>”: Group1: TP53 methylation higher 50%, Group2: TP53 methylation lower 50%</li>
</ul>
<p><strong>S3: Analyze &amp; Visualize</strong></p>
<ul>
<li>“<strong>S3.1 Set analysis parameters</strong>”: Log-rank text</li>
<li>“<strong>S3.2 Set visualization parameters</strong>”: Modify color; Set confidence interval.</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1310"></span>
<img src="images/p1310.png" alt="Example of survival analysis in individual mode" width="100%" />
<p class="caption">
Figure 12.10: Example of survival analysis in individual mode
</p>
</div>
</div>
<div id="pan-cancer-mode-2" class="section level3 hasAnchor" number="12.4.2">
<h3><span class="header-section-number">12.4.2</span> Pan-cancer Mode<a href="per-tpc-2.html#pan-cancer-mode-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Perform the survival analysis between two groups of samples generated by <strong>one identifier</strong> across <strong>multiple tumor</strong>.</p>
<p>The following is a simple example. (The steps not mentioned in above figure imply default operations.)</p>
<p><strong>S1: Preset</strong></p>
<ul>
<li>“<strong>S1.2 Choose cancer</strong>”: choose the <strong><em>BRCA</em></strong> cancer type</li>
<li>“<strong>S1.3 Filter samples</strong>”: choose the <strong><em>non-normal</em></strong> samples via the quick filter.</li>
</ul>
<p><strong>S2: Get data</strong></p>
<ul>
<li>“<strong>S2.1 Select survival endpoint</strong>”: Select the OS event</li>
<li>“<strong>S2.2 Divide 2 groups by one condition</strong>”: Group1: Signature value higher 10%, Group2: Signature value lower 10%</li>
</ul>
<p><strong>S3: Analyze &amp; Visualize</strong></p>
<ul>
<li>“<strong>S3.1 Set analysis parameters</strong>”: Univariate Cox regression; Use the initial data.</li>
<li>“<strong>S3.2 Set visualization parameters</strong>”: Add main title.</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1311"></span>
<img src="images/p1311.png" alt="Example of survival analysis in pan-cancer mode" width="100%" />
<p class="caption">
Figure 12.11: Example of survival analysis in pan-cancer mode
</p>
</div>
</div>
<div id="batch-screen-mode-2" class="section level3 hasAnchor" number="12.4.3">
<h3><span class="header-section-number">12.4.3</span> Batch Screen Mode<a href="per-tpc-2.html#batch-screen-mode-2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Perform the survival analysis between two groups of samples generated by <strong>multiple identifiers</strong> from <strong>one tumor</strong>.</p>
<p>The following is a simple example. (The steps not mentioned in above figure imply default operations.)</p>
<p><strong>S1: Preset</strong></p>
<ul>
<li>“<strong>S1.2 Choose cancer</strong>”: choose the all TCGA cancer types</li>
<li>“<strong>S1.3 Filter samples</strong>”: choose the <strong><em>primary tumor</em></strong> samples via the quick filter.</li>
<li>“<strong>S1.5 Add signature</strong>”: Design a molecular signature and add to custom metadata.</li>
</ul>
<p><strong>S2: Get data</strong></p>
<ul>
<li>“<strong>S2.1 Select survival endpoint</strong>”: Select the PFI event</li>
<li>“<strong>S2.2 Divide 2 groups by one condition</strong>”: Pathway activity→HALLMARK pathways→Group1: score higher 50%, Group2: score lower 50%</li>
</ul>
<p><strong>S3: Analyze</strong></p>
<ul>
<li>“<strong>S3.1 Set analysis parameters</strong>”: Log-rank test</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1312"></span>
<img src="images/p1312.png" alt="Example of survival analysis in batch screen mode" width="100%" />
<p class="caption">
Figure 12.12: Example of survival analysis in batch screen mode
</p>
</div>
</div>
</div>
<div id="cross-omics-analysis" class="section level2 hasAnchor" number="12.5">
<h2><span class="header-section-number">12.5</span> Cross-Omics analysis<a href="per-tpc-2.html#cross-omics-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Recently, we have added one pipeline for TCGA cross-omics analysis to simultaneously explore the molecular features of multiple omics across pan-cancers.</p>
<div id="gene-cross-omics-analysis" class="section level3 hasAnchor" number="12.5.1">
<h3><span class="header-section-number">12.5.1</span> Gene Cross-Omics Analysis<a href="per-tpc-2.html#gene-cross-omics-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Perform the cross-omics analysis at the gene level, involving Gene expression, Gene Mutation, Gene CNV, Transcript expression, DNA methylation.</p>
<p>The following is a simple example. (The steps not mentioned in above figure imply default operations.)</p>
<p><strong>S1: Preset</strong></p>
<ul>
<li>“<strong>S1.2 Choose cancer</strong>”: choose the all TCGA cancer types</li>
<li>“<strong>S1.3 Filter samples</strong>”: By default, select all normal samples and tumor samples.</li>
</ul>
<p><strong>S2: Get data</strong></p>
<ul>
<li>“<strong>S2.1 Select one gene</strong>”: Select gene TP53</li>
<li>“<strong>S2.2 Load mRNA/Mutation/CNV data</strong>”: Preload the mRNA/Mutation/CNV data of gene to save time in S3 step.</li>
<li>“<strong>S2.3 Load transcript data</strong>”: Select and load several transcript of gene, filter invalid transcripts with null value.</li>
<li>“<strong>S2.4 Load methylation data</strong>”: Select several CpG sits of gene and load methyaltion data.</li>
</ul>
<p><strong>S3: Analyze &amp; Visualize</strong> : Analyze multi-omics data and visualize by <code>funkyheatmap</code> package.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1313"></span>
<img src="images/p1313.png" alt="Example of Gene Cross-Omics Analysis" width="100%" />
<p class="caption">
Figure 12.13: Example of Gene Cross-Omics Analysis
</p>
</div>
<p><strong>Column-1</strong>: TCGA Names;</p>
<p><strong>Column-2[Expression Profile]</strong>: 0-1 normalized median expression of gene in normal samples (include GTEx);</p>
<p><strong>Column-3[Expression Profile]</strong>: 0-1 normalized median expression of gene in tumor samples;</p>
<p><strong>Column-4[Expression Profile]</strong>: The significance symbol of gene comparison between tumor and normal samples via <em>Wilcoxon</em> test (P&lt;0.001, “***”; P&lt;0.01, “**”; P&lt;0.05, “*”; P&gt;0.05, “-”).</p>
<p><strong>Column-5[Mutation Profile]</strong>: The distribution of gene mutation or wild status in tumor samples.</p>
<p><strong>Column-6[Mutation Profile]</strong>: The percentage of gene mutation status in tumor samples.</p>
<p><strong>Column-7[CNV Profile]</strong>: The distribution of gene copy number variation status in tumor samples (-2, homozygous deletion; -1, single copy deletion; 0, diploid normal copy; 1: low-level copy number amplification; 2: high-level copy number amplification).</p>
<p><strong>Column-8[CNV Profile]</strong>: The percentage of gene copy number amplification (1,2) in tumor samples.</p>
<p><strong>Column-9[CNV Profile]</strong>: The percentage of gene copy number deletion (-1, -2) in tumor samples.</p>
<p><strong>Column-Selected transcript(s) of one gene[CNV Profile]</strong>: The median transcript expression in tumor samples.</p>
<p><strong>Column-Selected CpG sites of one gene[Methylation Profile]</strong>: The median beta value of CpG sites in tumor samples.</p>
</div>
<div id="pathway-cross-omics-analysis" class="section level3 hasAnchor" number="12.5.2">
<h3><span class="header-section-number">12.5.2</span> Pathway Cross-Omics Analysis<a href="per-tpc-2.html#pathway-cross-omics-analysis" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Perform the cross-omics analysis at the pathway level, involving Gene expression, Gene Mutation, Gene CNV.</p>
<p>The following is a simple example. (The steps not mentioned in above figure imply default operations.)</p>
<p><strong>S1: Preset</strong></p>
<ul>
<li>“<strong>S1.2 Choose cancer</strong>”: choose the all TCGA cancer types</li>
<li>“<strong>S1.3 Filter samples</strong>”: By default, select all normal samples and tumor samples.</li>
</ul>
<p><strong>S2: Get data</strong></p>
<ul>
<li>“<strong>S2.1 one pathway</strong>”: Select pathway HALLMARK_ADIPOGENESIS</li>
</ul>
<p><strong>S3: Analyze &amp; Visualize</strong> : Analyze multi-omics data and visualize by <code>funkyheatmap</code> package.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1314"></span>
<img src="images/p1314.png" alt="Example of Pathway Cross-Omics Analysis" width="100%" />
<p class="caption">
Figure 12.14: Example of Pathway Cross-Omics Analysis
</p>
</div>
<p><strong>Column-1</strong>: TCGA Names;</p>
<p><strong>Column-2[Expression Profile]</strong>: 0-1 normalized median ssGSEA score of pathway in normal samples (include GTEx);</p>
<p><strong>Column-3[Expression Profile]</strong>: 0-1 normalized median ssGSEA score of gene in tumor samples;</p>
<p><strong>Column-4[Expression Profile]</strong>: The significance symbol of gene comparison between tumor and normal samples via <em>Wilcoxon</em> test (P&lt;0.001, “***”; P&lt;0.01, “**”; P&lt;0.05, “*”; P&gt;0.05, “-”).</p>
<p><strong>Column-5[Mutation Profile]</strong>: The distribution of mutation or wild pathway gene status in tumor samples. If one of pathway genes is mutated, it is considered as mutation.</p>
<p><strong>Column-6[Mutation Profile]</strong>: The percentage of pathway mutation status in tumor samples.</p>
<p><strong>Column-7[Mutation Profile]</strong>: The mean counts of mutated pathway genes in tumor samples.</p>
<p><strong>Column-8[CNV Profile]</strong>: The distribution of pathway gene copy number variation status in tumor samples. (Amp, homozygous deletion or single copy deletion; Non, diploid normal copy; Del, Copy number amplification.)</p>
<p><strong>Column-9[CNV Profile]</strong>: The copy number amplification percentage of pathway genes in tumor samples.</p>
<p><strong>Column-10[CNV Profile]</strong>: The copy number deletion percentage of pathway gene in tumor samples.</p>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="per-tpc-1.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="download.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/13-app-personalTPC-2.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
