<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 11 Personalized TPC pipelines(1) | UCSCXenaShiny v2 Book</title>
  <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="generator" content="bookdown 0.37 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 11 Personalized TPC pipelines(1) | UCSCXenaShiny v2 Book" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 11 Personalized TPC pipelines(1) | UCSCXenaShiny v2 Book" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::gitbook,
set in the _output.yml file.</p>" />
  

<meta name="author" content="Shensuo Li, Shixiang Wang" />


<meta name="date" content="2024-07-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="custom-tpc.html"/>
<link rel="next" href="per-tpc-2.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>About</a></li>
<li class="chapter" data-level="1" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="intro.html"><a href="intro.html#ucsc-xena-datasets"><i class="fa fa-check"></i><b>1.1</b> UCSC Xena Datasets</a></li>
<li class="chapter" data-level="1.2" data-path="intro.html"><a href="intro.html#tcga-abbreviations"><i class="fa fa-check"></i><b>1.2</b> TCGA abbreviations</a></li>
<li class="chapter" data-level="1.3" data-path="intro.html"><a href="intro.html#pcawg-abbreviations"><i class="fa fa-check"></i><b>1.3</b> PCAWG abbreviations</a></li>
</ul></li>
<li class="part"><span><b>I R package</b></span></li>
<li class="chapter" data-level="2" data-path="install.html"><a href="install.html"><i class="fa fa-check"></i><b>2</b> Install</a>
<ul>
<li class="chapter" data-level="2.1" data-path="install.html"><a href="install.html#cran"><i class="fa fa-check"></i><b>2.1</b> CRAN</a></li>
<li class="chapter" data-level="2.2" data-path="install.html"><a href="install.html#github"><i class="fa fa-check"></i><b>2.2</b> Github</a></li>
<li class="chapter" data-level="2.3" data-path="install.html"><a href="install.html#conda"><i class="fa fa-check"></i><b>2.3</b> Conda</a></li>
<li class="chapter" data-level="2.4" data-path="install.html"><a href="install.html#docker"><i class="fa fa-check"></i><b>2.4</b> Docker</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="build-in.html"><a href="build-in.html"><i class="fa fa-check"></i><b>3</b> Built-in datasets</a>
<ul>
<li class="chapter" data-level="3.1" data-path="build-in.html"><a href="build-in.html#tcga"><i class="fa fa-check"></i><b>3.1</b> TCGA</a></li>
<li class="chapter" data-level="3.2" data-path="build-in.html"><a href="build-in.html#pcawg"><i class="fa fa-check"></i><b>3.2</b> PCAWG</a></li>
<li class="chapter" data-level="3.3" data-path="build-in.html"><a href="build-in.html#ccle"><i class="fa fa-check"></i><b>3.3</b> CCLE</a></li>
<li class="chapter" data-level="3.4" data-path="build-in.html"><a href="build-in.html#zenodo"><i class="fa fa-check"></i><b>3.4</b> Zenodo</a>
<ul>
<li class="chapter" data-level="3.4.1" data-path="build-in.html"><a href="build-in.html#tumor-infiltration-estimations"><i class="fa fa-check"></i><b>3.4.1</b> tumor infiltration estimations</a></li>
<li class="chapter" data-level="3.4.2" data-path="build-in.html"><a href="build-in.html#ssgsea-pathway-activities"><i class="fa fa-check"></i><b>3.4.2</b> ssGSEA pathway activities</a></li>
<li class="chapter" data-level="3.4.3" data-path="build-in.html"><a href="build-in.html#other-tcga-annotations"><i class="fa fa-check"></i><b>3.4.3</b> other TCGA annotations</a></li>
<li class="chapter" data-level="3.4.4" data-path="build-in.html"><a href="build-in.html#identifier-repository"><i class="fa fa-check"></i><b>3.4.4</b> identifier repository</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="4" data-path="data-query.html"><a href="data-query.html"><i class="fa fa-check"></i><b>4</b> Molecular data query</a>
<ul>
<li class="chapter" data-level="4.1" data-path="data-query.html"><a href="data-query.html#tcga-query"><i class="fa fa-check"></i><b>4.1</b> TCGA query</a>
<ul>
<li class="chapter" data-level="4.1.1" data-path="data-query.html"><a href="data-query.html#get_pancan_gene_value"><i class="fa fa-check"></i><b>4.1.1</b> get_pancan_gene_value()</a></li>
<li class="chapter" data-level="4.1.2" data-path="data-query.html"><a href="data-query.html#get_pancan_transcript_value"><i class="fa fa-check"></i><b>4.1.2</b> get_pancan_transcript_value()</a></li>
<li class="chapter" data-level="4.1.3" data-path="data-query.html"><a href="data-query.html#get_pancan_protein_value"><i class="fa fa-check"></i><b>4.1.3</b> get_pancan_protein_value()</a></li>
<li class="chapter" data-level="4.1.4" data-path="data-query.html"><a href="data-query.html#get_pancan_mutation_status"><i class="fa fa-check"></i><b>4.1.4</b> get_pancan_mutation_status()</a></li>
<li class="chapter" data-level="4.1.5" data-path="data-query.html"><a href="data-query.html#get_pancan_cn_value"><i class="fa fa-check"></i><b>4.1.5</b> get_pancan_cn_value()</a></li>
<li class="chapter" data-level="4.1.6" data-path="data-query.html"><a href="data-query.html#get_pancan_methylation_value"><i class="fa fa-check"></i><b>4.1.6</b> get_pancan_methylation_value()</a></li>
<li class="chapter" data-level="4.1.7" data-path="data-query.html"><a href="data-query.html#get_pancan_mirna_value"><i class="fa fa-check"></i><b>4.1.7</b> get_pancan_miRNA_value()</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="data-query.html"><a href="data-query.html#pcawg-query"><i class="fa fa-check"></i><b>4.2</b> PCAWG query</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="data-query.html"><a href="data-query.html#get_pcawg_gene_value"><i class="fa fa-check"></i><b>4.2.1</b> get_pcawg_gene_value()</a></li>
<li class="chapter" data-level="4.2.2" data-path="data-query.html"><a href="data-query.html#get_pcawg_fusion_value"><i class="fa fa-check"></i><b>4.2.2</b> get_pcawg_fusion_value()</a></li>
<li class="chapter" data-level="4.2.3" data-path="data-query.html"><a href="data-query.html#get_pcawg_mirna_value"><i class="fa fa-check"></i><b>4.2.3</b> get_pcawg_miRNA_value()</a></li>
<li class="chapter" data-level="4.2.4" data-path="data-query.html"><a href="data-query.html#get_pcawg_promoter_value"><i class="fa fa-check"></i><b>4.2.4</b> get_pcawg_promoter_value()</a></li>
<li class="chapter" data-level="4.2.5" data-path="data-query.html"><a href="data-query.html#get_pcawg_apobec_mutagenesis_value"><i class="fa fa-check"></i><b>4.2.5</b> get_pcawg_APOBEC_mutagenesis_value()</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="data-query.html"><a href="data-query.html#ccle-query"><i class="fa fa-check"></i><b>4.3</b> CCLE query</a>
<ul>
<li class="chapter" data-level="4.3.1" data-path="data-query.html"><a href="data-query.html#get_ccle_gene_value"><i class="fa fa-check"></i><b>4.3.1</b> get_ccle_gene_value()</a></li>
<li class="chapter" data-level="4.3.2" data-path="data-query.html"><a href="data-query.html#get_ccle_protein_value"><i class="fa fa-check"></i><b>4.3.2</b> get_ccle_protein_value()</a></li>
<li class="chapter" data-level="4.3.3" data-path="data-query.html"><a href="data-query.html#get_ccle_mutation_status"><i class="fa fa-check"></i><b>4.3.3</b> get_ccle_mutation_status()</a></li>
<li class="chapter" data-level="4.3.4" data-path="data-query.html"><a href="data-query.html#get_ccle_cn_value"><i class="fa fa-check"></i><b>4.3.4</b> get_ccle_cn_value()</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="data-query.html"><a href="data-query.html#general-query"><i class="fa fa-check"></i><b>4.4</b> General query</a>
<ul>
<li class="chapter" data-level="4.4.1" data-path="data-query.html"><a href="data-query.html#query_pancan_value"><i class="fa fa-check"></i><b>4.4.1</b> query_pancan_value()</a></li>
<li class="chapter" data-level="4.4.2" data-path="data-query.html"><a href="data-query.html#query_molecule_value"><i class="fa fa-check"></i><b>4.4.2</b> query_molecule_value()</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="data-analysis.html"><a href="data-analysis.html"><i class="fa fa-check"></i><b>5</b> Molecular data analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="data-analysis.html"><a href="data-analysis.html#tcga-analysis"><i class="fa fa-check"></i><b>5.1</b> TCGA analysis</a>
<ul>
<li class="chapter" data-level="5.1.1" data-path="data-analysis.html"><a href="data-analysis.html#comparison-analysis"><i class="fa fa-check"></i><b>5.1.1</b> Comparison analysis</a></li>
<li class="chapter" data-level="5.1.2" data-path="data-analysis.html"><a href="data-analysis.html#correlation-analysis"><i class="fa fa-check"></i><b>5.1.2</b> Correlation analysis</a></li>
<li class="chapter" data-level="5.1.3" data-path="data-analysis.html"><a href="data-analysis.html#survival-analysis"><i class="fa fa-check"></i><b>5.1.3</b> Survival analysis</a></li>
<li class="chapter" data-level="5.1.4" data-path="data-analysis.html"><a href="data-analysis.html#dimension-reduction"><i class="fa fa-check"></i><b>5.1.4</b> Dimension reduction</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="data-analysis.html"><a href="data-analysis.html#pcawg-analysis"><i class="fa fa-check"></i><b>5.2</b> PCAWG analysis</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="data-analysis.html"><a href="data-analysis.html#comparsion-analysis"><i class="fa fa-check"></i><b>5.2.1</b> Comparsion analysis</a></li>
<li class="chapter" data-level="5.2.2" data-path="data-analysis.html"><a href="data-analysis.html#correlation-analysis-1"><i class="fa fa-check"></i><b>5.2.2</b> Correlation analysis</a></li>
<li class="chapter" data-level="5.2.3" data-path="data-analysis.html"><a href="data-analysis.html#survival-analysis-1"><i class="fa fa-check"></i><b>5.2.3</b> Survival analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="data-analysis.html"><a href="data-analysis.html#ccle-analysis"><i class="fa fa-check"></i><b>5.3</b> CCLE analysis</a>
<ul>
<li class="chapter" data-level="5.3.1" data-path="data-analysis.html"><a href="data-analysis.html#comparsion-analysis-1"><i class="fa fa-check"></i><b>5.3.1</b> Comparsion analysis</a></li>
<li class="chapter" data-level="5.3.2" data-path="data-analysis.html"><a href="data-analysis.html#correlation-analysis-2"><i class="fa fa-check"></i><b>5.3.2</b> Correlation analysis</a></li>
</ul></li>
<li class="chapter" data-level="5.4" data-path="data-analysis.html"><a href="data-analysis.html#general-analysis"><i class="fa fa-check"></i><b>5.4</b> General analysis</a>
<ul>
<li class="chapter" data-level="5.4.1" data-path="data-analysis.html"><a href="data-analysis.html#comparison-analysis-1"><i class="fa fa-check"></i><b>5.4.1</b> Comparison analysis</a></li>
<li class="chapter" data-level="5.4.2" data-path="data-analysis.html"><a href="data-analysis.html#correlation-analysis-3"><i class="fa fa-check"></i><b>5.4.2</b> Correlation analysis</a></li>
<li class="chapter" data-level="5.4.3" data-path="data-analysis.html"><a href="data-analysis.html#survival-analysis-2"><i class="fa fa-check"></i><b>5.4.3</b> Survival analysis</a></li>
<li class="chapter" data-level="5.4.4" data-path="data-analysis.html"><a href="data-analysis.html#dimension-reduction-1"><i class="fa fa-check"></i><b>5.4.4</b> Dimension reduction</a></li>
</ul></li>
</ul></li>
<li class="part"><span><b>II Shiny App</b></span></li>
<li class="chapter" data-level="6" data-path="visit.html"><a href="visit.html"><i class="fa fa-check"></i><b>6</b> Visit</a>
<ul>
<li class="chapter" data-level="6.1" data-path="visit.html"><a href="visit.html#server-deployments"><i class="fa fa-check"></i><b>6.1</b> Server deployments</a></li>
<li class="chapter" data-level="6.2" data-path="visit.html"><a href="visit.html#local-deployment"><i class="fa fa-check"></i><b>6.2</b> Local deployment</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="home.html"><a href="home.html"><i class="fa fa-check"></i><b>7</b> Homepage</a>
<ul>
<li class="chapter" data-level="7.1" data-path="home.html"><a href="home.html#quick-molecule-exploration"><i class="fa fa-check"></i><b>7.1</b> Quick Molecule Exploration</a>
<ul>
<li class="chapter" data-level="7.1.1" data-path="home.html"><a href="home.html#daily-gene"><i class="fa fa-check"></i><b>7.1.1</b> Daily Gene</a></li>
<li class="chapter" data-level="7.1.2" data-path="home.html"><a href="home.html#pan-cancer-query"><i class="fa fa-check"></i><b>7.1.2</b> Pan-cancer Query</a></li>
</ul></li>
<li class="chapter" data-level="7.2" data-path="home.html"><a href="home.html#slick-gallery"><i class="fa fa-check"></i><b>7.2</b> Slick Gallery</a></li>
<li class="chapter" data-level="7.3" data-path="home.html"><a href="home.html#tcga-analysis-guide"><i class="fa fa-check"></i><b>7.3</b> TCGA Analysis Guide</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="repo.html"><a href="repo.html"><i class="fa fa-check"></i><b>8</b> Repository</a>
<ul>
<li class="chapter" data-level="8.1" data-path="repo.html"><a href="repo.html#show-metadata"><i class="fa fa-check"></i><b>8.1</b> Show Metadata</a></li>
<li class="chapter" data-level="8.2" data-path="repo.html"><a href="repo.html#request-data"><i class="fa fa-check"></i><b>8.2</b> Request Data</a></li>
<li class="chapter" data-level="8.3" data-path="repo.html"><a href="repo.html#analyze-data"><i class="fa fa-check"></i><b>8.3</b> Analyze Data</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="general-da.html"><a href="general-da.html"><i class="fa fa-check"></i><b>9</b> General Dataset Analysis</a>
<ul>
<li class="chapter" data-level="9.1" data-path="general-da.html"><a href="general-da.html#m-vis-identifier-cor"><i class="fa fa-check"></i><b>9.1</b> Scatter-Correlation</a></li>
<li class="chapter" data-level="9.2" data-path="general-da.html"><a href="general-da.html#m-vis-identifier-multi-cor"><i class="fa fa-check"></i><b>9.2</b> Matrix-Correlation</a></li>
<li class="chapter" data-level="9.3" data-path="general-da.html"><a href="general-da.html#m-vis-identifier-grp-comparison"><i class="fa fa-check"></i><b>9.3</b> Group-Comparison</a></li>
<li class="chapter" data-level="9.4" data-path="general-da.html"><a href="general-da.html#m-vis-identifier-grp-surv"><i class="fa fa-check"></i><b>9.4</b> Survival-Analysis</a></li>
<li class="chapter" data-level="9.5" data-path="general-da.html"><a href="general-da.html#m-vis-identifier-dim-dist"><i class="fa fa-check"></i><b>9.5</b> Dimension-Distribution</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="custom-tpc.html"><a href="custom-tpc.html"><i class="fa fa-check"></i><b>10</b> Custom TPC modules</a>
<ul>
<li class="chapter" data-level="10.1" data-path="custom-tpc.html"><a href="custom-tpc.html#tcga-1"><i class="fa fa-check"></i><b>10.1</b> TCGA</a>
<ul>
<li class="chapter" data-level="10.1.1" data-path="custom-tpc.html"><a href="custom-tpc.html#comparison"><i class="fa fa-check"></i><b>10.1.1</b> Comparison</a></li>
<li class="chapter" data-level="10.1.2" data-path="custom-tpc.html"><a href="custom-tpc.html#correlation"><i class="fa fa-check"></i><b>10.1.2</b> Correlation</a></li>
<li class="chapter" data-level="10.1.3" data-path="custom-tpc.html"><a href="custom-tpc.html#survival-analysis-3"><i class="fa fa-check"></i><b>10.1.3</b> Survival analysis</a></li>
<li class="chapter" data-level="10.1.4" data-path="custom-tpc.html"><a href="custom-tpc.html#dimension-reduction-2"><i class="fa fa-check"></i><b>10.1.4</b> Dimension Reduction</a></li>
</ul></li>
<li class="chapter" data-level="10.2" data-path="custom-tpc.html"><a href="custom-tpc.html#pcawg-1"><i class="fa fa-check"></i><b>10.2</b> PCAWG</a>
<ul>
<li class="chapter" data-level="10.2.1" data-path="custom-tpc.html"><a href="custom-tpc.html#comparison-analysis-2"><i class="fa fa-check"></i><b>10.2.1</b> Comparison analysis</a></li>
<li class="chapter" data-level="10.2.2" data-path="custom-tpc.html"><a href="custom-tpc.html#correlation-analysis-4"><i class="fa fa-check"></i><b>10.2.2</b> Correlation analysis</a></li>
<li class="chapter" data-level="10.2.3" data-path="custom-tpc.html"><a href="custom-tpc.html#survival-analysis-4"><i class="fa fa-check"></i><b>10.2.3</b> Survival analysis</a></li>
</ul></li>
<li class="chapter" data-level="10.3" data-path="custom-tpc.html"><a href="custom-tpc.html#ccle-1"><i class="fa fa-check"></i><b>10.3</b> CCLE</a>
<ul>
<li class="chapter" data-level="10.3.1" data-path="custom-tpc.html"><a href="custom-tpc.html#comparison-analysis-3"><i class="fa fa-check"></i><b>10.3.1</b> Comparison analysis</a></li>
<li class="chapter" data-level="10.3.2" data-path="custom-tpc.html"><a href="custom-tpc.html#correlation-analysis-5"><i class="fa fa-check"></i><b>10.3.2</b> Correlation analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="11" data-path="per-tpc-1.html"><a href="per-tpc-1.html"><i class="fa fa-check"></i><b>11</b> Personalized TPC pipelines(1)</a>
<ul>
<li class="chapter" data-level="11.1" data-path="per-tpc-1.html"><a href="per-tpc-1.html#extensive-tumor-data"><i class="fa fa-check"></i><b>11.1</b> Extensive tumor data</a>
<ul>
<li class="chapter" data-level="11.1.1" data-path="per-tpc-1.html"><a href="per-tpc-1.html#compiling-types-of-identifiers"><i class="fa fa-check"></i><b>11.1.1</b> Compiling types of identifiers</a></li>
<li class="chapter" data-level="11.1.2" data-path="per-tpc-1.html"><a href="per-tpc-1.html#reference-of-full-identifiers"><i class="fa fa-check"></i><b>11.1.2</b> Reference of full identifiers</a></li>
<li class="chapter" data-level="11.1.3" data-path="per-tpc-1.html"><a href="per-tpc-1.html#user-defined-identifiers"><i class="fa fa-check"></i><b>11.1.3</b> User-defined identifiers</a></li>
</ul></li>
<li class="chapter" data-level="11.2" data-path="per-tpc-1.html"><a href="per-tpc-1.html#sample-processing"><i class="fa fa-check"></i><b>11.2</b> Sample processing</a>
<ul>
<li class="chapter" data-level="11.2.1" data-path="per-tpc-1.html"><a href="per-tpc-1.html#sample-filtering"><i class="fa fa-check"></i><b>11.2.1</b> Sample filtering</a></li>
<li class="chapter" data-level="11.2.2" data-path="per-tpc-1.html"><a href="per-tpc-1.html#sample-grouping"><i class="fa fa-check"></i><b>11.2.2</b> Sample grouping</a></li>
</ul></li>
<li class="chapter" data-level="11.3" data-path="per-tpc-1.html"><a href="per-tpc-1.html#data-fetch"><i class="fa fa-check"></i><b>11.3</b> Data fetch</a>
<ul>
<li class="chapter" data-level="11.3.1" data-path="per-tpc-1.html"><a href="per-tpc-1.html#single-data"><i class="fa fa-check"></i><b>11.3.1</b> Single data</a></li>
<li class="chapter" data-level="11.3.2" data-path="per-tpc-1.html"><a href="per-tpc-1.html#multiple-data"><i class="fa fa-check"></i><b>11.3.2</b> Multiple data</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="12" data-path="per-tpc-2.html"><a href="per-tpc-2.html"><i class="fa fa-check"></i><b>12</b> Personalized TPC pipelines(2)</a>
<ul>
<li class="chapter" data-level="12.1" data-path="per-tpc-2.html"><a href="per-tpc-2.html#general-steps"><i class="fa fa-check"></i><b>12.1</b> General steps</a></li>
<li class="chapter" data-level="12.2" data-path="per-tpc-2.html"><a href="per-tpc-2.html#correlation-analysis-6"><i class="fa fa-check"></i><b>12.2</b> Correlation analysis</a>
<ul>
<li class="chapter" data-level="12.2.1" data-path="per-tpc-2.html"><a href="per-tpc-2.html#individual-mode"><i class="fa fa-check"></i><b>12.2.1</b> Individual Mode</a></li>
<li class="chapter" data-level="12.2.2" data-path="per-tpc-2.html"><a href="per-tpc-2.html#pan-cancer-mode"><i class="fa fa-check"></i><b>12.2.2</b> Pan-cancer Mode</a></li>
<li class="chapter" data-level="12.2.3" data-path="per-tpc-2.html"><a href="per-tpc-2.html#batch-screen-mode"><i class="fa fa-check"></i><b>12.2.3</b> Batch Screen Mode</a></li>
</ul></li>
<li class="chapter" data-level="12.3" data-path="per-tpc-2.html"><a href="per-tpc-2.html#comparison-analysis-4"><i class="fa fa-check"></i><b>12.3</b> Comparison analysis</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="per-tpc-2.html"><a href="per-tpc-2.html#individual-mode-1"><i class="fa fa-check"></i><b>12.3.1</b> Individual Mode</a></li>
<li class="chapter" data-level="12.3.2" data-path="per-tpc-2.html"><a href="per-tpc-2.html#pan-cancer-mode-1"><i class="fa fa-check"></i><b>12.3.2</b> Pan-cancer Mode</a></li>
<li class="chapter" data-level="12.3.3" data-path="per-tpc-2.html"><a href="per-tpc-2.html#batch-screen-mode-1"><i class="fa fa-check"></i><b>12.3.3</b> Batch Screen Mode</a></li>
</ul></li>
<li class="chapter" data-level="12.4" data-path="per-tpc-2.html"><a href="per-tpc-2.html#survival-analysis-5"><i class="fa fa-check"></i><b>12.4</b> Survival analysis</a>
<ul>
<li class="chapter" data-level="12.4.1" data-path="per-tpc-2.html"><a href="per-tpc-2.html#individual-mode-2"><i class="fa fa-check"></i><b>12.4.1</b> Individual Mode</a></li>
<li class="chapter" data-level="12.4.2" data-path="per-tpc-2.html"><a href="per-tpc-2.html#pan-cancer-mode-2"><i class="fa fa-check"></i><b>12.4.2</b> Pan-cancer Mode</a></li>
<li class="chapter" data-level="12.4.3" data-path="per-tpc-2.html"><a href="per-tpc-2.html#batch-screen-mode-2"><i class="fa fa-check"></i><b>12.4.3</b> Batch Screen Mode</a></li>
</ul></li>
<li class="chapter" data-level="12.5" data-path="per-tpc-2.html"><a href="per-tpc-2.html#cross-omics-analysis"><i class="fa fa-check"></i><b>12.5</b> Cross-Omics analysis</a>
<ul>
<li class="chapter" data-level="12.5.1" data-path="per-tpc-2.html"><a href="per-tpc-2.html#gene-cross-omics-analysis"><i class="fa fa-check"></i><b>12.5.1</b> Gene Cross-Omics Analysis</a></li>
<li class="chapter" data-level="12.5.2" data-path="per-tpc-2.html"><a href="per-tpc-2.html#pathway-cross-omics-analysis"><i class="fa fa-check"></i><b>12.5.2</b> Pathway Cross-Omics Analysis</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="13" data-path="download.html"><a href="download.html"><i class="fa fa-check"></i><b>13</b> Download modules</a>
<ul>
<li class="chapter" data-level="13.1" data-path="download.html"><a href="download.html#download-tpc-datasets"><i class="fa fa-check"></i><b>13.1</b> Download TPC datasets</a>
<ul>
<li class="chapter" data-level="13.1.1" data-path="download.html"><a href="download.html#omics-molecular-data"><i class="fa fa-check"></i><b>13.1.1</b> Omics molecular data</a></li>
<li class="chapter" data-level="13.1.2" data-path="download.html"><a href="download.html#non-omics-annotation-data"><i class="fa fa-check"></i><b>13.1.2</b> Non-omics annotation data</a></li>
</ul></li>
<li class="chapter" data-level="13.2" data-path="download.html"><a href="download.html#download-ucsc-xena-datasets"><i class="fa fa-check"></i><b>13.2</b> Download UCSC Xena datasets</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/openbiox/UCSCXenaShiny" target="blank">UCSCXenaShiny Github</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">UCSCXenaShiny v2 Book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="per-tpc-1" class="section level1 hasAnchor" number="11">
<h1><span class="header-section-number">Chapter 11</span> Personalized TPC pipelines(1)<a href="per-tpc-1.html#per-tpc-1" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="extensive-tumor-data" class="section level2 hasAnchor" number="11.1">
<h2><span class="header-section-number">11.1</span> Extensive tumor data<a href="per-tpc-1.html#extensive-tumor-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="compiling-types-of-identifiers" class="section level3 hasAnchor" number="11.1.1">
<h3><span class="header-section-number">11.1.1</span> Compiling types of identifiers<a href="per-tpc-1.html#compiling-types-of-identifiers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ol style="list-style-type: decimal">
<li>For <strong>TCGA</strong> samples, there are <strong>5</strong> main data types and <strong>23</strong> subtypes (except for <em>Custom metadata</em>).</li>
</ol>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1201"></span>
<img src="images/p1201.png" alt="Hierarchical types of TCGA Identifiers" width="100%" />
<p class="caption">
Figure 11.1: Hierarchical types of TCGA Identifiers
</p>
</div>
<ol start="2" style="list-style-type: decimal">
<li><p>For <strong>PCAWG</strong> samples, there are <strong>5</strong> main data types and <strong>17</strong> subtypes (except for <em>Custom metadata</em>).</p></li>
<li><p>For <strong>CCLE</strong> samples, there are <strong>3</strong> main data types and <strong>6</strong> subtypes (except for <em>Custom metadata</em>).</p></li>
</ol>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1202"></span>
<img src="images/p1202.png" alt="Hierarchical types of PCAWG and CCLE Identifiers" width="100%" />
<p class="caption">
Figure 11.2: Hierarchical types of PCAWG and CCLE Identifiers
</p>
</div>
<p>About the core data type —— <strong>Molecular profile</strong>, the initial step of all analysis pipelines involves the check and modification of molecular datasets through the <code>Modify datasets[opt]</code> widget. For some molecular types of TPC databases, there are alternative datasets for users to select, which greatly enriches the analytical possibilities</p>
<ul>
<li>Among the 7 types of TCGA molecules, 4 of them have the alternative datasets.</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1102"></span>
<img src="images/p1102.png" alt="Alternative datasets of TCGA molecules" width="100%" />
<p class="caption">
Figure 11.3: Alternative datasets of TCGA molecules
</p>
</div>
<blockquote>
<p>Tips: For DNA Methylation, mean value of all CpG sites under one gene will be calculated for the gene by default. In the <code>Modify datasets[opt]</code> widget, users can furtherly limit its CpG sites and modify the aggregation method (e.g. median).</p>
</blockquote>
<ul>
<li><p>Among the 5 types of PCAWG molecules, 2 of them have the alternative datasets.</p></li>
<li><p>Among the 4 types of CCLE molecules, 1 of them has the alternative datasets.</p></li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1103"></span>
<img src="images/p1103.png" alt="Alternative datasets of PCAWG and CCLE molecules" width="100%" />
<p class="caption">
Figure 11.4: Alternative datasets of PCAWG and CCLE molecules
</p>
</div>
</div>
<div id="reference-of-full-identifiers" class="section level3 hasAnchor" number="11.1.2">
<h3><span class="header-section-number">11.1.2</span> Reference of full identifiers<a href="per-tpc-1.html#reference-of-full-identifiers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<ul>
<li>In the “<strong>Help→ID query</strong>” page, all identifiers for “<em>Molecular profile</em>”, “<em>Tumor index</em>”, “<em>Immune Infiltration</em>”, “<em>Pathway activity</em>”, “<em>Phenotype data</em>” types of TPC samples are compiling for easy query before analysis.</li>
<li>For example, valid identifiers under the <em>TCGA mRNA Expression</em> subtype are shown in the following figure.</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1203"></span>
<img src="images/p1203.png" alt="TPC Identifiers Query Page" width="100%" />
<p class="caption">
Figure 11.5: TPC Identifiers Query Page
</p>
</div>
</div>
<div id="user-defined-identifiers" class="section level3 hasAnchor" number="11.1.3">
<h3><span class="header-section-number">11.1.3</span> User-defined identifiers<a href="per-tpc-1.html#user-defined-identifiers" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div id="upload-of-random-identifiers" class="section level4 hasAnchor" number="11.1.3.1">
<h4><span class="header-section-number">11.1.3.1</span> Upload of random identifiers<a href="per-tpc-1.html#upload-of-random-identifiers" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ol style="list-style-type: decimal">
<li>Users are enabled to upload random identifiers for joint analysis in the “<strong>Upload metadata</strong>” part of <strong>S1</strong> step.</li>
<li>The example file can be downloaded via the right button to see the format requirements.
<ul>
<li>.CSV file;</li>
<li>The first column names “Sample” with sample ids (e.g. “TCGA-OR-A5J1-01” for TCGA samples). Only the intersected samples between this column and all TPC samples are considered;</li>
<li>The other columns are random identifiers with continuous or categorical values.</li>
</ul></li>
<li>Upon successful identifiers upload, they will be added into the “Custom metadata” subtype of “Phenotype data”.</li>
</ol>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1204"></span>
<img src="images/p1204.png" alt="Upload of random identifiers" width="80%" />
<p class="caption">
Figure 11.6: Upload of random identifiers
</p>
</div>
</div>
<div id="design-of-custom-signature" class="section level4 hasAnchor" number="11.1.3.2">
<h4><span class="header-section-number">11.1.3.2</span> Design of custom signature<a href="per-tpc-1.html#design-of-custom-signature" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<ol style="list-style-type: decimal">
<li>Users can also design custom signature score based on self-defined molecular formula in the “<strong>Add signature</strong>” part of <strong>S1</strong> step.</li>
<li>Upon successful signature design and score calculation, they will also be added into the “Custom metadata” subtype of “Phenotype data”.</li>
</ol>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1205"></span>
<img src="images/p1205.png" alt="Design of custom signature" width="80%" />
<p class="caption">
Figure 11.7: Design of custom signature
</p>
</div>
<ol start="3" style="list-style-type: decimal">
<li>As shown in the figure below, the signature can be set as following steps after clicking Edit button.</li>
</ol>
<ul>
<li>Firstly, modify the name and select the molecular type;</li>
<li>Then, design the molecular formula comprised of multiple molecules with respective coefficients;
<ul>
<li>The coefficient of each molecule is divided into two parts (Directions and Absolute Coef).</li>
<li>The “<strong>Del</strong>” button will cancel the last set molecule and the “<strong>Reset</strong>” button will undo all selected molecules.</li>
<li>The real-time formula is shown in the bottom.</li>
</ul></li>
<li>Finally, click the “Query data” button to calculate the signature scores and click the right button to add the signature into the “Custom metadata” subtype of “Phenotype data”. In addition, user can obtain the result locally via the bottom button.</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1206"></span>
<img src="images/p1206.png" alt="3 steps of signature design" width="100%" />
<p class="caption">
Figure 11.8: 3 steps of signature design
</p>
</div>
</div>
</div>
</div>
<div id="sample-processing" class="section level2 hasAnchor" number="11.2">
<h2><span class="header-section-number">11.2</span> Sample processing<a href="per-tpc-1.html#sample-processing" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1207"></span>
<img src="images/p1207.png" alt="Sample filtering and grouping modules" width="80%" />
<p class="caption">
Figure 11.9: Sample filtering and grouping modules
</p>
</div>
<div id="sample-filtering" class="section level3 hasAnchor" number="11.2.1">
<h3><span class="header-section-number">11.2.1</span> Sample filtering<a href="per-tpc-1.html#sample-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>After selecting one tumor, all its samples will be selected by default. Here, we provide two methods to obtain specific subpopulation in the “<strong>Filter samples</strong>” part of <strong>S1</strong> step.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1208"></span>
<img src="images/p1208.png" alt="Sample filtering module" width="80%" />
<p class="caption">
Figure 11.10: Sample filtering module
</p>
</div>
<div id="quick-filter" class="section level4 hasAnchor" number="11.2.1.1">
<h4><span class="header-section-number">11.2.1.1</span> Quick filter<a href="per-tpc-1.html#quick-filter" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Users can directly make quick selections based on tissue code (pathological) types of samples. Based on the <a href="https://gdc.cancer.gov/resources-tcga-users/tcga-code-tables/sample-type-codes">14th to 15th characters</a> of the TCGA sample ID (e.g., TCGA-19-1787-01), samples can be categorized into different types. The common code types involved in UCSCXenaShiny analysis are shown in the following table.</p>
<table>
<colgroup>
<col width="5%" />
<col width="69%" />
<col width="25%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Code</th>
<th align="left">Definition</th>
<th align="left">Short Letter Code</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">01</td>
<td align="left">Primary Solid Tumor</td>
<td align="left">TP</td>
</tr>
<tr class="even">
<td align="left">02</td>
<td align="left">Recurrent Solid Tumor</td>
<td align="left">TR</td>
</tr>
<tr class="odd">
<td align="left">03</td>
<td align="left">Primary Blood Derived Cancer - Peripheral Blood</td>
<td align="left">TB</td>
</tr>
<tr class="even">
<td align="left">05</td>
<td align="left">Additional - New Primary</td>
<td align="left">TAP</td>
</tr>
<tr class="odd">
<td align="left">06</td>
<td align="left">Metastatic</td>
<td align="left">TM</td>
</tr>
<tr class="even">
<td align="left">07</td>
<td align="left">Additional Metastatic</td>
<td align="left">TAM</td>
</tr>
<tr class="odd">
<td align="left">11</td>
<td align="left">Solid Tissue Normal</td>
<td align="left">NT</td>
</tr>
</tbody>
</table>
<blockquote>
<p>For PCAWG database, you can perform similar operation to quickly select tumor or normal samples. For CCLE database, there is no quick filter.</p>
</blockquote>
</div>
<div id="exact-filter" class="section level4 hasAnchor" number="11.2.1.2">
<h4><span class="header-section-number">11.2.1.2</span> Exact filter<a href="per-tpc-1.html#exact-filter" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<p>Through this module, users are enabled to perform personalized sample filtering. This followings are operational steps:</p>
<ol style="list-style-type: decimal">
<li>Set the candidate conditions from comprehensive identifiers. Each addition will rename the selected identifier as “Condition_Num”.
<ul>
<li>When selecting molecular identifier as condition, it may take some time to download from the USCS Xena, depending on the network.</li>
<li>For TPC samples, some basic clinical variables have been selected into the candidate pool. For TCGA samples, they include <em>Age, Code, Gender, Stage_ajcc, Stage_clinical, Grade</em>.</li>
</ul></li>
<li>Get an overview of the distribution of each candidate condition for clear filtering .
<ul>
<li>For continuous variables, the mean and the distribution of multiple percentiles are displayed.</li>
<li>For categorical variables, the frequency of each category is displayed.</li>
</ul></li>
<li>Perform filtering operation based on one or multiple conditions from the above candidate list. Total 6 operators are devised for versatile filtering give the type of each condition.
<ul>
<li>For categorical variable, “<strong>+</strong>” and “<strong>-</strong>” are used for retain or discard specific categories of samples.</li>
<li>For continuous variable, “<strong>&gt;</strong>” and “<strong>&lt;</strong>” are used to set absolute cutoff, while “<strong>%&gt;</strong>” and “<strong>%&lt;</strong>” are used to set percentile cutoff.</li>
</ul></li>
</ol>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1209"></span>
<img src="images/p1209.png" alt="3 steps of sample filtering" width="100%" />
<p class="caption">
Figure 11.11: 3 steps of sample filtering
</p>
</div>
<blockquote>
<p>Tips: If the percentile cutoff is involved, the results may vary based on the order of filtering, as illustrated in the example above figure.</p>
<ul>
<li>The first instance: Samples with the top 50% high expression of TP53 in female samples. → <strong>23</strong> samples</li>
<li>The second instance: Females from the top 50% of samples with high TP53 expression. → <strong>28</strong> samples</li>
</ul>
</blockquote>
</div>
</div>
<div id="sample-grouping" class="section level3 hasAnchor" number="11.2.2">
<h3><span class="header-section-number">11.2.2</span> Sample grouping<a href="per-tpc-1.html#sample-grouping" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For binary comparison or survival analysis, it is necessary to divide the samples into two groups based on one specific condition. Here, we also provide a personalized grouping module to flexibly set two ranges of sample populations.</p>
<p>Generally, it has the two main steps:</p>
<ol style="list-style-type: decimal">
<li><p>Select one condition and observe its distribution. Similar to the filtering module, it can bring a clear pre-knowledge by observing the distribution of one continuous or categorical condition.</p></li>
<li><p>Set two non-overlapping sample ranges to group.</p></li>
</ol>
<ul>
<li>For categorical variable, assign one or more classifications for both groups (<strong>Left figure</strong>)</li>
<li>For continuous variable, set the minimum and max thresholds for both group.
<ul>
<li>By default, the first group (Group1) has lower value with <strong>left close</strong> and <strong>right open</strong> range, e.g. [0, 1). If its left range is blank, it will represent the minimum value of the condition.</li>
<li>And, the second group (Group2) has larger value with <strong>left close</strong> and <strong>right close</strong> range, e.g. [0, 1]. If its right range is blank, it will represent the maximum value of the condition.</li>
<li>The thresholds can be percentile values (<strong>Middle figure</strong>) or absolute values (<strong>Right figure</strong>) depending on purposes.</li>
</ul></li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1210"></span>
<img src="images/p1210.png" alt="Sample grouping module" width="100%" />
<p class="caption">
Figure 11.12: Sample grouping module
</p>
</div>
<blockquote>
<p>Tips: By default, the Group1 has the prior order during analysis and visualization. You can click the switch “Whether reverse levels” to change the order.</p>
</blockquote>
</div>
</div>
<div id="data-fetch" class="section level2 hasAnchor" number="11.3">
<h2><span class="header-section-number">11.3</span> Data fetch<a href="per-tpc-1.html#data-fetch" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>After sample selection and processing, we usually need to fetch data for downstream analysis.</p>
<div id="single-data" class="section level3 hasAnchor" number="11.3.1">
<h3><span class="header-section-number">11.3.1</span> Single data<a href="per-tpc-1.html#single-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Most of time, we just need to select one identifier and fetch its values. According to the hierarchical organization (<strong>Data type → Data subtype → Identifier</strong>), select one interesting identifier and then click the “Query button” to fetch data. It’s worth noting that, for <em>Molecular profile</em> data types, it may take some time to download from the UCSC Xena, depending on the network.</p>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1211"></span>
<img src="images/p1211.png" alt="Fetch single data" width="80%" />
<p class="caption">
Figure 11.13: Fetch single data
</p>
</div>
</div>
<div id="multiple-data" class="section level3 hasAnchor" number="11.3.2">
<h3><span class="header-section-number">11.3.2</span> Multiple data<a href="per-tpc-1.html#multiple-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>For batch analysis, we can select multiple identifiers based on one of three ways.</p>
<ol style="list-style-type: decimal">
<li>In the first way (<strong>Left figure</strong>), user can select batch identifiers of one specific subtype one by one.</li>
<li>In the second way (<strong>Middle figure</strong>), user can select all identifiers of one specific subtype. Notably, for some molecular types with much identifiers, all the identifiers (gene/transcript) belong to one MSigDB signature can be selected.</li>
</ol>
<ul>
<li>These molecular types are “mRNA Expression”, “DNA Methylation”, “Mutation status”, “Copy Number Variation”, “Transcript Expression”.</li>
<li>The MSigDB gene signatures are obtained via <a href="https://igordot.github.io/msigdbr/"><code>msigdbr</code> package</a>, which can be 9 main categories (bottom table).</li>
<li>Two links are available for detailed information of selected signature and MSigDB database.</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>In the third way (<strong>Right figure</strong>), user can directly upload an identifier file with following requirements:</li>
</ol>
<ul>
<li>TXT format;</li>
<li>One column without colname;</li>
<li>Only the valid identifiers that overlapped with the selected data subtype are retained</li>
</ul>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:p1212"></span>
<img src="images/p1212.png" alt="Fetch multiple data" width="100%" />
<p class="caption">
Figure 11.14: Fetch multiple data
</p>
</div>
<hr />
<table>
<colgroup>
<col width="8%" />
<col width="31%" />
<col width="60%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Category</th>
<th align="center">Notes</th>
<th>Subcategory</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">H</td>
<td align="center">hallmark gene sets</td>
<td>/</td>
</tr>
<tr class="even">
<td align="center">C1</td>
<td align="center">positional gene sets</td>
<td>/</td>
</tr>
<tr class="odd">
<td align="center">C2</td>
<td align="center">curated gene sets</td>
<td>CGP, CP, CP:BIOCARTA, CP:KEGG, CP:PID, CP:REACTOME, CP:WIKIPATHWAYS</td>
</tr>
<tr class="even">
<td align="center">C3</td>
<td align="center">regulatory target gene sets</td>
<td>MIR:MIRDB, MIR:MIR_Legacy, TFT:GTRD, TFT:TFT_Legacy</td>
</tr>
<tr class="odd">
<td align="center">C4</td>
<td align="center">computational gene sets</td>
<td>CGN, CM</td>
</tr>
<tr class="even">
<td align="center">C5</td>
<td align="center">ontology gene sets</td>
<td><a href="GO:BP" class="uri">GO:BP</a>, <a href="GO:CC" class="uri">GO:CC</a>, <a href="GO:MF" class="uri">GO:MF</a>, HPO</td>
</tr>
<tr class="odd">
<td align="center">C6</td>
<td align="center">oncogenic signature gene sets</td>
<td>/</td>
</tr>
<tr class="even">
<td align="center">C7</td>
<td align="center">immunologic signature gene sets</td>
<td>IMMUNESIGDB, VAX</td>
</tr>
<tr class="odd">
<td align="center">C8</td>
<td align="center">cell type signature gene sets</td>
<td>/</td>
</tr>
</tbody>
</table>

</div>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="custom-tpc.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="per-tpc-2.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/12-app-personalTPC-1.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
}
});
});
</script>

</body>

</html>
